From 5b763c172986f05cc1ea2dc94140e7aa604945b2 Mon Sep 17 00:00:00 2001
From: John Audia <graysky@archlinux.us>
Date: Wed, 16 Feb 2022 09:46:00 -0500
Subject: [PATCH 1/2] kernel: bump 5.10 to 5.10.101

Removed upstreamed:
  pending-5.10/841-USB-serial-option-add-ZTE-MF286D-modem.patch[1]
  bcm27xx/950-0592-drm-vc4-Allow-DBLCLK-modes-even-if-horz-timing-is-od.patch[2]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.101&id=7113440a36c741efd7c76e3d70b3634100120cdb
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.101&id=21c890ca8eaecea06cabb92be2a53a6f26f56383

Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200

Signed-off-by: John Audia <graysky@archlinux.us>
---
 include/kernel-5.10                           |  4 +-
 ...cec_available-flag-as-always-support.patch |  4 +-
 .../950-0313-vc4_hdmi-Set-HDMI_MAI_FMT.patch  |  6 +-
 ...-vc4-add-iec958-controls-to-vc4_hdmi.patch |  6 +-
 ...move-setup-from-hw_params-to-prepare.patch |  6 +-
 ...enable-HBR-MAI-format-on-HBR-streams.patch |  2 +-
 ...firmware-logic-for-MAI-threshold-set.patch |  2 +-
 ...t-VC4_HDMI_MAI_CONFIG_FORMAT_REVERSE.patch |  2 +-
 ...tft-Add-support-for-display-variants.patch |  2 +-
 ...resent-but-empty-dmas-disables-audio.patch |  2 +-
 ...the-HDMI-registers-into-the-debugfs-.patch |  4 +-
 ...re-pixel-frequency-in-the-connector-.patch |  2 +-
 ...drm-vc4-hdmi-Enable-10-12-bpc-output.patch |  2 +-
 ...7-drm-vc4-hdmi-Introduce-a-CEC-clock.patch |  4 +-
 ...c4-hdmi-Split-the-interrupt-handlers.patch |  6 +-
 ...-Support-BCM2711-CEC-interrupt-setup.patch | 10 ++--
 ...-t-register-the-CEC-adapter-if-there.patch |  2 +-
 ...metadata-property-to-the-VC5-HDMI-co.patch |  6 +-
 ...rm-Avoid-full-hdmi-audio-fifo-writes.patch |  2 +-
 ...c4-hdmi-Raise-the-maximum-clock-rate.patch |  2 +-
 ...LCLK-modes-even-if-horz-timing-is-od.patch | 42 --------------
 ...-interface-in-mode6-for-18-bit-color.patch |  4 +-
 ...efiles-dt-Always-set-on-ARCH_BCM2835.patch |  2 +-
 ...-support-BCM4908-s-integrated-switch.patch |  8 +--
 ..._sf2-setup-BCM4908-internal-crossbar.patch | 10 ++--
 ...-bcm_sf2-Fill-in-BCM4908-CFP-entries.patch |  2 +-
 ...-add-function-finding-RGMII-register.patch | 14 ++---
 ...-dsa-bcm_sf2-fix-BCM4908-RGMII-reg-s.patch |  2 +-
 ...Fix-bcm_sf2_reg_rgmii_cntrl-call-for.patch |  4 +-
 ...dsa-bcm_sf2-refactor-LED-regs-access.patch |  2 +-
 ...m_sf2-enable-GPHY-for-switch-probing.patch |  2 +-
 ...sf2-keep-GPHY-enabled-on-the-BCM4908.patch |  2 +-
 ...phy-marvell-refactor-HWMON-OOP-style.patch | 28 ++++-----
 ...-marvell-add-SFP-support-for-88E1510.patch |  4 +-
 ...equest-assisted-learning-on-CPU-port.patch |  2 +-
 ...B-serial-option-add-ZTE-MF286D-modem.patch | 57 -------------------
 .../901-arm-boot-add-dts-files.patch          |  2 +-
 .../0069-arm-boot-add-dts-files.patch         |  2 +-
 ...-lantiq-enable-jumbo-frames-on-GSWIP.patch |  8 +--
 .../062-add-sun8i-h3-zeropi-support.patch     |  2 +-
 ...0-sunxi-h3-add-support-for-nanopi-r1.patch |  2 +-
 41 files changed, 89 insertions(+), 188 deletions(-)
 delete mode 100644 target/linux/bcm27xx/patches-5.10/950-0592-drm-vc4-Allow-DBLCLK-modes-even-if-horz-timing-is-od.patch
 delete mode 100644 target/linux/generic/pending-5.10/841-USB-serial-option-add-ZTE-MF286D-modem.patch

diff --git a/include/kernel-5.10 b/include/kernel-5.10
index c71d0deac274..47c2fd41f5b3 100644
--- a/include/kernel-5.10
+++ b/include/kernel-5.10
@@ -1,2 +1,2 @@
-LINUX_VERSION-5.10 = .100
-LINUX_KERNEL_HASH-5.10.100 = d56965afc9b6a3d26d53db40ccd37fd9d15f2ca6bfd54ef6f0f8b6e92c170999
+LINUX_VERSION-5.10 = .101
+LINUX_KERNEL_HASH-5.10.101 = 0749258cb7ee47c478dfc34857e8930d41bfe0b074e80e9f57cbfa56d0ba1c8e
diff --git a/target/linux/bcm27xx/patches-5.10/950-0310-vc4_hdmi-Remove-cec_available-flag-as-always-support.patch b/target/linux/bcm27xx/patches-5.10/950-0310-vc4_hdmi-Remove-cec_available-flag-as-always-support.patch
index d7610cabb0c3..4c9e0d71f4bd 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0310-vc4_hdmi-Remove-cec_available-flag-as-always-support.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0310-vc4_hdmi-Remove-cec_available-flag-as-always-support.patch
@@ -12,7 +12,7 @@ Signed-off-by: Dom Cobley <popcornmix@gmail.com>
 
 --- a/drivers/gpu/drm/vc4/vc4_hdmi.c
 +++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
-@@ -1491,9 +1491,6 @@ static int vc4_hdmi_cec_init(struct vc4_
+@@ -1493,9 +1493,6 @@ static int vc4_hdmi_cec_init(struct vc4_
  	u32 value;
  	int ret;
  
@@ -22,7 +22,7 @@ Signed-off-by: Dom Cobley <popcornmix@gmail.com>
  	vc4_hdmi->cec_adap = cec_allocate_adapter(&vc4_hdmi_cec_adap_ops,
  						  vc4_hdmi, "vc4",
  						  CEC_CAP_DEFAULTS |
-@@ -1926,7 +1923,6 @@ static const struct vc4_hdmi_variant bcm
+@@ -1928,7 +1925,6 @@ static const struct vc4_hdmi_variant bcm
  	.debugfs_name		= "hdmi_regs",
  	.card_name		= "vc4-hdmi",
  	.max_pixel_clock	= 162000000,
diff --git a/target/linux/bcm27xx/patches-5.10/950-0313-vc4_hdmi-Set-HDMI_MAI_FMT.patch b/target/linux/bcm27xx/patches-5.10/950-0313-vc4_hdmi-Set-HDMI_MAI_FMT.patch
index 3cebccc100ed..e4f6f8795c36 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0313-vc4_hdmi-Set-HDMI_MAI_FMT.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0313-vc4_hdmi-Set-HDMI_MAI_FMT.patch
@@ -14,7 +14,7 @@ Signed-off-by: Dom Cobley <popcornmix@gmail.com>
 
 --- a/drivers/gpu/drm/vc4/vc4_hdmi.c
 +++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
-@@ -989,6 +989,44 @@ static void vc4_hdmi_audio_shutdown(stru
+@@ -991,6 +991,44 @@ static void vc4_hdmi_audio_shutdown(stru
  	vc4_hdmi->audio.substream = NULL;
  }
  
@@ -59,7 +59,7 @@ Signed-off-by: Dom Cobley <popcornmix@gmail.com>
  /* HDMI audio codec callbacks */
  static int vc4_hdmi_audio_hw_params(struct snd_pcm_substream *substream,
  				    struct snd_pcm_hw_params *params,
-@@ -999,6 +1037,8 @@ static int vc4_hdmi_audio_hw_params(stru
+@@ -1001,6 +1039,8 @@ static int vc4_hdmi_audio_hw_params(stru
  	struct device *dev = &vc4_hdmi->pdev->dev;
  	u32 audio_packet_config, channel_mask;
  	u32 channel_map;
@@ -68,7 +68,7 @@ Signed-off-by: Dom Cobley <popcornmix@gmail.com>
  
  	if (substream != vc4_hdmi->audio.substream)
  		return -EINVAL;
-@@ -1019,6 +1059,14 @@ static int vc4_hdmi_audio_hw_params(stru
+@@ -1021,6 +1061,14 @@ static int vc4_hdmi_audio_hw_params(stru
  
  	vc4_hdmi_audio_set_mai_clock(vc4_hdmi);
  
diff --git a/target/linux/bcm27xx/patches-5.10/950-0314-drm-vc4-add-iec958-controls-to-vc4_hdmi.patch b/target/linux/bcm27xx/patches-5.10/950-0314-drm-vc4-add-iec958-controls-to-vc4_hdmi.patch
index c5caa739b26e..9582c86871af 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0314-drm-vc4-add-iec958-controls-to-vc4_hdmi.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0314-drm-vc4-add-iec958-controls-to-vc4_hdmi.patch
@@ -26,7 +26,7 @@ Signed-off-by: Matthias Reichl <hias@horus.com>
  #include <sound/dmaengine_pcm.h>
  #include <sound/pcm_drm_eld.h>
  #include <sound/pcm_params.h>
-@@ -1180,6 +1181,47 @@ static int vc4_hdmi_audio_eld_ctl_get(st
+@@ -1182,6 +1183,47 @@ static int vc4_hdmi_audio_eld_ctl_get(st
  	return 0;
  }
  
@@ -74,7 +74,7 @@ Signed-off-by: Matthias Reichl <hias@horus.com>
  static const struct snd_kcontrol_new vc4_hdmi_audio_controls[] = {
  	{
  		.access = SNDRV_CTL_ELEM_ACCESS_READ |
-@@ -1189,6 +1231,19 @@ static const struct snd_kcontrol_new vc4
+@@ -1191,6 +1233,19 @@ static const struct snd_kcontrol_new vc4
  		.info = vc4_hdmi_audio_eld_ctl_info,
  		.get = vc4_hdmi_audio_eld_ctl_get,
  	},
@@ -94,7 +94,7 @@ Signed-off-by: Matthias Reichl <hias@horus.com>
  };
  
  static const struct snd_soc_dapm_widget vc4_hdmi_audio_widgets[] = {
-@@ -1309,6 +1364,11 @@ static int vc4_hdmi_audio_init(struct vc
+@@ -1311,6 +1366,11 @@ static int vc4_hdmi_audio_init(struct vc
  	vc4_hdmi->audio.dma_data.addr_width = DMA_SLAVE_BUSWIDTH_4_BYTES;
  	vc4_hdmi->audio.dma_data.maxburst = 2;
  
diff --git a/target/linux/bcm27xx/patches-5.10/950-0315-drm-vc4-move-setup-from-hw_params-to-prepare.patch b/target/linux/bcm27xx/patches-5.10/950-0315-drm-vc4-move-setup-from-hw_params-to-prepare.patch
index 109718017627..9b90426393c6 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0315-drm-vc4-move-setup-from-hw_params-to-prepare.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0315-drm-vc4-move-setup-from-hw_params-to-prepare.patch
@@ -14,7 +14,7 @@ Signed-off-by: Matthias Reichl <hias@horus.com>
 
 --- a/drivers/gpu/drm/vc4/vc4_hdmi.c
 +++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
-@@ -1029,9 +1029,8 @@ static int sample_rate_to_mai_fmt(int sa
+@@ -1031,9 +1031,8 @@ static int sample_rate_to_mai_fmt(int sa
  }
  
  /* HDMI audio codec callbacks */
@@ -26,7 +26,7 @@ Signed-off-by: Matthias Reichl <hias@horus.com>
  {
  	struct vc4_hdmi *vc4_hdmi = dai_to_hdmi(dai);
  	struct drm_encoder *encoder = &vc4_hdmi->encoder.base.base;
-@@ -1044,12 +1043,15 @@ static int vc4_hdmi_audio_hw_params(stru
+@@ -1046,12 +1045,15 @@ static int vc4_hdmi_audio_hw_params(stru
  	if (substream != vc4_hdmi->audio.substream)
  		return -EINVAL;
  
@@ -47,7 +47,7 @@ Signed-off-by: Matthias Reichl <hias@horus.com>
  
  	HDMI_WRITE(HDMI_MAI_CTL,
  		   VC4_HD_MAI_CTL_RESET |
-@@ -1271,7 +1273,7 @@ static const struct snd_soc_component_dr
+@@ -1273,7 +1275,7 @@ static const struct snd_soc_component_dr
  static const struct snd_soc_dai_ops vc4_hdmi_audio_dai_ops = {
  	.startup = vc4_hdmi_audio_startup,
  	.shutdown = vc4_hdmi_audio_shutdown,
diff --git a/target/linux/bcm27xx/patches-5.10/950-0316-drm-vc4-enable-HBR-MAI-format-on-HBR-streams.patch b/target/linux/bcm27xx/patches-5.10/950-0316-drm-vc4-enable-HBR-MAI-format-on-HBR-streams.patch
index b700930c8186..1fbda2013749 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0316-drm-vc4-enable-HBR-MAI-format-on-HBR-streams.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0316-drm-vc4-enable-HBR-MAI-format-on-HBR-streams.patch
@@ -10,7 +10,7 @@ Signed-off-by: Matthias Reichl <hias@horus.com>
 
 --- a/drivers/gpu/drm/vc4/vc4_hdmi.c
 +++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
-@@ -1063,7 +1063,11 @@ static int vc4_hdmi_audio_prepare(struct
+@@ -1065,7 +1065,11 @@ static int vc4_hdmi_audio_prepare(struct
  	vc4_hdmi_audio_set_mai_clock(vc4_hdmi);
  
  	mai_sample_rate = sample_rate_to_mai_fmt(vc4_hdmi->audio.samplerate);
diff --git a/target/linux/bcm27xx/patches-5.10/950-0317-vc4_hdmi-Remove-firmware-logic-for-MAI-threshold-set.patch b/target/linux/bcm27xx/patches-5.10/950-0317-vc4_hdmi-Remove-firmware-logic-for-MAI-threshold-set.patch
index 8f16329b04ba..ef776face17b 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0317-vc4_hdmi-Remove-firmware-logic-for-MAI-threshold-set.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0317-vc4_hdmi-Remove-firmware-logic-for-MAI-threshold-set.patch
@@ -16,7 +16,7 @@ Signed-off-by: Dom Cobley <popcornmix@gmail.com>
 
 --- a/drivers/gpu/drm/vc4/vc4_hdmi.c
 +++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
-@@ -1084,22 +1084,12 @@ static int vc4_hdmi_audio_prepare(struct
+@@ -1086,22 +1086,12 @@ static int vc4_hdmi_audio_prepare(struct
  	audio_packet_config |= VC4_SET_FIELD(channel_mask,
  					     VC4_HDMI_AUDIO_PACKET_CEA_MASK);
  
diff --git a/target/linux/bcm27xx/patches-5.10/950-0318-vc_hdmi-Set-VC4_HDMI_MAI_CONFIG_FORMAT_REVERSE.patch b/target/linux/bcm27xx/patches-5.10/950-0318-vc_hdmi-Set-VC4_HDMI_MAI_CONFIG_FORMAT_REVERSE.patch
index 6f09dac58868..1767ccf4a5a6 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0318-vc_hdmi-Set-VC4_HDMI_MAI_CONFIG_FORMAT_REVERSE.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0318-vc_hdmi-Set-VC4_HDMI_MAI_CONFIG_FORMAT_REVERSE.patch
@@ -14,7 +14,7 @@ Signed-off-by: Dom Cobley <popcornmix@gmail.com>
 
 --- a/drivers/gpu/drm/vc4/vc4_hdmi.c
 +++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
-@@ -1093,6 +1093,7 @@ static int vc4_hdmi_audio_prepare(struct
+@@ -1095,6 +1095,7 @@ static int vc4_hdmi_audio_prepare(struct
  
  	HDMI_WRITE(HDMI_MAI_CONFIG,
  		   VC4_HDMI_MAI_CONFIG_BIT_REVERSE |
diff --git a/target/linux/bcm27xx/patches-5.10/950-0332-staging-fbtft-Add-support-for-display-variants.patch b/target/linux/bcm27xx/patches-5.10/950-0332-staging-fbtft-Add-support-for-display-variants.patch
index 7bdc80b46cce..9bb513628cd2 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0332-staging-fbtft-Add-support-for-display-variants.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0332-staging-fbtft-Add-support-for-display-variants.patch
@@ -191,7 +191,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  	{},                                                                \
  };                                                                         \
  									   \
-@@ -344,6 +355,11 @@ static void __exit fbtft_driver_module_e
+@@ -347,6 +358,11 @@ static void __exit fbtft_driver_module_e
  module_init(fbtft_driver_module_init);                                     \
  module_exit(fbtft_driver_module_exit);
  
diff --git a/target/linux/bcm27xx/patches-5.10/950-0346-drm-vc4-A-present-but-empty-dmas-disables-audio.patch b/target/linux/bcm27xx/patches-5.10/950-0346-drm-vc4-A-present-but-empty-dmas-disables-audio.patch
index 0c104b178843..373f98cbd91b 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0346-drm-vc4-A-present-but-empty-dmas-disables-audio.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0346-drm-vc4-A-present-but-empty-dmas-disables-audio.patch
@@ -16,7 +16,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.org>
 
 --- a/drivers/gpu/drm/vc4/vc4_hdmi.c
 +++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
-@@ -1331,10 +1331,12 @@ static int vc4_hdmi_audio_init(struct vc
+@@ -1333,10 +1333,12 @@ static int vc4_hdmi_audio_init(struct vc
  	const __be32 *addr;
  	int index;
  	int ret;
diff --git a/target/linux/bcm27xx/patches-5.10/950-0349-drm-vc4-Add-all-the-HDMI-registers-into-the-debugfs-.patch b/target/linux/bcm27xx/patches-5.10/950-0349-drm-vc4-Add-all-the-HDMI-registers-into-the-debugfs-.patch
index edc5be80ab86..4a4c53570623 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0349-drm-vc4-Add-all-the-HDMI-registers-into-the-debugfs-.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0349-drm-vc4-Add-all-the-HDMI-registers-into-the-debugfs-.patch
@@ -29,7 +29,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  
  	return 0;
  }
-@@ -1732,6 +1738,7 @@ static int vc5_hdmi_init_resources(struc
+@@ -1734,6 +1740,7 @@ static int vc5_hdmi_init_resources(struc
  	struct platform_device *pdev = vc4_hdmi->pdev;
  	struct device *dev = &pdev->dev;
  	struct resource *res;
@@ -37,7 +37,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  
  	res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "hdmi");
  	if (!res)
-@@ -1822,6 +1829,38 @@ static int vc5_hdmi_init_resources(struc
+@@ -1824,6 +1831,38 @@ static int vc5_hdmi_init_resources(struc
  		return PTR_ERR(vc4_hdmi->reset);
  	}
  
diff --git a/target/linux/bcm27xx/patches-5.10/950-0415-drm-vc4-hdmi-Store-pixel-frequency-in-the-connector-.patch b/target/linux/bcm27xx/patches-5.10/950-0415-drm-vc4-hdmi-Store-pixel-frequency-in-the-connector-.patch
index 5e51b008f3b0..719f9235e541 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0415-drm-vc4-hdmi-Store-pixel-frequency-in-the-connector-.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0415-drm-vc4-hdmi-Store-pixel-frequency-in-the-connector-.patch
@@ -76,7 +76,7 @@ Signed-off-by: Maxime Ripard <maxime@cerno.tech>
  	struct drm_display_mode *mode = &crtc_state->adjusted_mode;
  	struct vc4_hdmi *vc4_hdmi = encoder_to_vc4_hdmi(encoder);
  	unsigned long long pixel_rate = mode->clock * 1000;
-@@ -875,6 +897,8 @@ static int vc4_hdmi_encoder_atomic_check
+@@ -876,6 +898,8 @@ static int vc4_hdmi_encoder_atomic_check
  	if (pixel_rate > vc4_hdmi->variant->max_pixel_clock)
  		return -EINVAL;
  
diff --git a/target/linux/bcm27xx/patches-5.10/950-0417-drm-vc4-hdmi-Enable-10-12-bpc-output.patch b/target/linux/bcm27xx/patches-5.10/950-0417-drm-vc4-hdmi-Enable-10-12-bpc-output.patch
index 69f255627f98..646475fa801a 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0417-drm-vc4-hdmi-Enable-10-12-bpc-output.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0417-drm-vc4-hdmi-Enable-10-12-bpc-output.patch
@@ -141,7 +141,7 @@ Reviewed-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  }
  
  static void vc4_hdmi_encoder_pre_crtc_enable(struct drm_encoder *encoder,
-@@ -891,6 +951,14 @@ static int vc4_hdmi_encoder_atomic_check
+@@ -892,6 +952,14 @@ static int vc4_hdmi_encoder_atomic_check
  		pixel_rate = mode->clock * 1000;
  	}
  
diff --git a/target/linux/bcm27xx/patches-5.10/950-0457-drm-vc4-hdmi-Introduce-a-CEC-clock.patch b/target/linux/bcm27xx/patches-5.10/950-0457-drm-vc4-hdmi-Introduce-a-CEC-clock.patch
index 36f46a71baf5..f20685c4559a 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0457-drm-vc4-hdmi-Introduce-a-CEC-clock.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0457-drm-vc4-hdmi-Introduce-a-CEC-clock.patch
@@ -27,7 +27,7 @@ Signed-off-by: Maxime Ripard <maxime@cerno.tech>
  	value |= clk_cnt << VC4_HDMI_CEC_DIV_CLK_CNT_SHIFT;
  	HDMI_WRITE(HDMI_CEC_CNTRL_1, value);
  }
-@@ -1867,6 +1867,7 @@ static int vc4_hdmi_init_resources(struc
+@@ -1869,6 +1869,7 @@ static int vc4_hdmi_init_resources(struc
  		return PTR_ERR(vc4_hdmi->hsm_clock);
  	}
  	vc4_hdmi->audio_clock = vc4_hdmi->hsm_clock;
@@ -35,7 +35,7 @@ Signed-off-by: Maxime Ripard <maxime@cerno.tech>
  
  	return 0;
  }
-@@ -1961,6 +1962,12 @@ static int vc5_hdmi_init_resources(struc
+@@ -1963,6 +1964,12 @@ static int vc5_hdmi_init_resources(struc
  		return PTR_ERR(vc4_hdmi->audio_clock);
  	}
  
diff --git a/target/linux/bcm27xx/patches-5.10/950-0458-drm-vc4-hdmi-Split-the-interrupt-handlers.patch b/target/linux/bcm27xx/patches-5.10/950-0458-drm-vc4-hdmi-Split-the-interrupt-handlers.patch
index 7e16acfacc85..5447b33259e8 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0458-drm-vc4-hdmi-Split-the-interrupt-handlers.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0458-drm-vc4-hdmi-Split-the-interrupt-handlers.patch
@@ -19,7 +19,7 @@ Reviewed-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
 
 --- a/drivers/gpu/drm/vc4/vc4_hdmi.c
 +++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
-@@ -1572,15 +1572,22 @@ static int vc4_hdmi_audio_init(struct vc
+@@ -1574,15 +1574,22 @@ static int vc4_hdmi_audio_init(struct vc
  }
  
  #ifdef CONFIG_DRM_VC4_HDMI_CEC
@@ -48,7 +48,7 @@ Reviewed-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  		cec_transmit_done(vc4_hdmi->cec_adap, CEC_TX_STATUS_OK,
  				  0, 0, 0, 0);
  	} else {
-@@ -1594,6 +1601,19 @@ static irqreturn_t vc4_cec_irq_handler_t
+@@ -1596,6 +1603,19 @@ static irqreturn_t vc4_cec_irq_handler_t
  	return IRQ_HANDLED;
  }
  
@@ -68,7 +68,7 @@ Reviewed-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  static void vc4_cec_read_msg(struct vc4_hdmi *vc4_hdmi, u32 cntrl1)
  {
  	struct drm_device *dev = vc4_hdmi->connector.dev;
-@@ -1618,31 +1638,55 @@ static void vc4_cec_read_msg(struct vc4_
+@@ -1620,31 +1640,55 @@ static void vc4_cec_read_msg(struct vc4_
  	}
  }
  
diff --git a/target/linux/bcm27xx/patches-5.10/950-0459-drm-vc4-hdmi-Support-BCM2711-CEC-interrupt-setup.patch b/target/linux/bcm27xx/patches-5.10/950-0459-drm-vc4-hdmi-Support-BCM2711-CEC-interrupt-setup.patch
index 39b80a2a2cd8..8c399b6c7951 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0459-drm-vc4-hdmi-Support-BCM2711-CEC-interrupt-setup.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0459-drm-vc4-hdmi-Support-BCM2711-CEC-interrupt-setup.patch
@@ -20,7 +20,7 @@ Reviewed-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
 
 --- a/drivers/gpu/drm/vc4/vc4_hdmi.c
 +++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
-@@ -1723,9 +1723,11 @@ static int vc4_hdmi_cec_adap_enable(stru
+@@ -1725,9 +1725,11 @@ static int vc4_hdmi_cec_adap_enable(stru
  			   ((3600 / usecs) << VC4_HDMI_CEC_CNT_TO_3600_US_SHIFT) |
  			   ((3500 / usecs) << VC4_HDMI_CEC_CNT_TO_3500_US_SHIFT));
  
@@ -34,7 +34,7 @@ Reviewed-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  		HDMI_WRITE(HDMI_CEC_CNTRL_5, val |
  			   VC4_HDMI_CEC_TX_SW_RESET | VC4_HDMI_CEC_RX_SW_RESET);
  	}
-@@ -1797,8 +1799,6 @@ static int vc4_hdmi_cec_init(struct vc4_
+@@ -1799,8 +1801,6 @@ static int vc4_hdmi_cec_init(struct vc4_
  	cec_fill_conn_info_from_drm(&conn_info, &vc4_hdmi->connector);
  	cec_s_conn_info(vc4_hdmi->cec_adap, &conn_info);
  
@@ -43,7 +43,7 @@ Reviewed-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  	value = HDMI_READ(HDMI_CEC_CNTRL_1);
  	/* Set the logical address to Unregistered */
  	value |= VC4_HDMI_CEC_ADDR_MASK;
-@@ -1806,12 +1806,32 @@ static int vc4_hdmi_cec_init(struct vc4_
+@@ -1808,12 +1808,32 @@ static int vc4_hdmi_cec_init(struct vc4_
  
  	vc4_hdmi_cec_update_clk_div(vc4_hdmi);
  
@@ -82,7 +82,7 @@ Reviewed-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  
  	ret = cec_register_adapter(vc4_hdmi->cec_adap, &pdev->dev);
  	if (ret < 0)
-@@ -2286,6 +2306,7 @@ static const struct vc4_hdmi_variant bcm
+@@ -2288,6 +2308,7 @@ static const struct vc4_hdmi_variant bcm
  		PHY_LANE_CK,
  	},
  	.unsupported_odd_h_timings	= true,
@@ -90,7 +90,7 @@ Reviewed-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  
  	.init_resources		= vc5_hdmi_init_resources,
  	.csc_setup		= vc5_hdmi_csc_setup,
-@@ -2312,6 +2333,7 @@ static const struct vc4_hdmi_variant bcm
+@@ -2314,6 +2335,7 @@ static const struct vc4_hdmi_variant bcm
  		PHY_LANE_2,
  	},
  	.unsupported_odd_h_timings	= true,
diff --git a/target/linux/bcm27xx/patches-5.10/950-0460-drm-vc4-hdmi-Don-t-register-the-CEC-adapter-if-there.patch b/target/linux/bcm27xx/patches-5.10/950-0460-drm-vc4-hdmi-Don-t-register-the-CEC-adapter-if-there.patch
index 2ead4e366711..cbace6303756 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0460-drm-vc4-hdmi-Don-t-register-the-CEC-adapter-if-there.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0460-drm-vc4-hdmi-Don-t-register-the-CEC-adapter-if-there.patch
@@ -19,7 +19,7 @@ Signed-off-by: Maxime Ripard <maxime@cerno.tech>
 
 --- a/drivers/gpu/drm/vc4/vc4_hdmi.c
 +++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
-@@ -1785,9 +1785,15 @@ static int vc4_hdmi_cec_init(struct vc4_
+@@ -1787,9 +1787,15 @@ static int vc4_hdmi_cec_init(struct vc4_
  {
  	struct cec_connector_info conn_info;
  	struct platform_device *pdev = vc4_hdmi->pdev;
diff --git a/target/linux/bcm27xx/patches-5.10/950-0475-drm-vc4-Add-HDR-metadata-property-to-the-VC5-HDMI-co.patch b/target/linux/bcm27xx/patches-5.10/950-0475-drm-vc4-Add-HDR-metadata-property-to-the-VC5-HDMI-co.patch
index 01bf5901aa88..554fffc09e31 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0475-drm-vc4-Add-HDR-metadata-property-to-the-VC5-HDMI-co.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0475-drm-vc4-Add-HDR-metadata-property-to-the-VC5-HDMI-co.patch
@@ -61,7 +61,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  }
  
  static void vc4_hdmi_encoder_post_crtc_disable(struct drm_encoder *encoder,
-@@ -2296,6 +2321,7 @@ static const struct vc4_hdmi_variant bcm
+@@ -2298,6 +2323,7 @@ static const struct vc4_hdmi_variant bcm
  	.phy_rng_enable		= vc4_hdmi_phy_rng_enable,
  	.phy_rng_disable	= vc4_hdmi_phy_rng_disable,
  	.channel_map		= vc4_hdmi_channel_map,
@@ -69,7 +69,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  };
  
  static const struct vc4_hdmi_variant bcm2711_hdmi0_variant = {
-@@ -2323,6 +2349,7 @@ static const struct vc4_hdmi_variant bcm
+@@ -2325,6 +2351,7 @@ static const struct vc4_hdmi_variant bcm
  	.phy_rng_enable		= vc5_hdmi_phy_rng_enable,
  	.phy_rng_disable	= vc5_hdmi_phy_rng_disable,
  	.channel_map		= vc5_hdmi_channel_map,
@@ -77,7 +77,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  };
  
  static const struct vc4_hdmi_variant bcm2711_hdmi1_variant = {
-@@ -2350,6 +2377,7 @@ static const struct vc4_hdmi_variant bcm
+@@ -2352,6 +2379,7 @@ static const struct vc4_hdmi_variant bcm
  	.phy_rng_enable		= vc5_hdmi_phy_rng_enable,
  	.phy_rng_disable	= vc5_hdmi_phy_rng_disable,
  	.channel_map		= vc5_hdmi_channel_map,
diff --git a/target/linux/bcm27xx/patches-5.10/950-0505-vc4-drm-Avoid-full-hdmi-audio-fifo-writes.patch b/target/linux/bcm27xx/patches-5.10/950-0505-vc4-drm-Avoid-full-hdmi-audio-fifo-writes.patch
index 98e477aecc19..4f89ad56e57a 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0505-vc4-drm-Avoid-full-hdmi-audio-fifo-writes.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0505-vc4-drm-Avoid-full-hdmi-audio-fifo-writes.patch
@@ -15,7 +15,7 @@ Signed-off-by: Dom Cobley <popcornmix@gmail.com>
 
 --- a/drivers/gpu/drm/vc4/vc4_hdmi.c
 +++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
-@@ -1295,10 +1295,10 @@ static int vc4_hdmi_audio_prepare(struct
+@@ -1297,10 +1297,10 @@ static int vc4_hdmi_audio_prepare(struct
  
  	/* Set the MAI threshold */
  	HDMI_WRITE(HDMI_MAI_THR,
diff --git a/target/linux/bcm27xx/patches-5.10/950-0576-drm-vc4-hdmi-Raise-the-maximum-clock-rate.patch b/target/linux/bcm27xx/patches-5.10/950-0576-drm-vc4-hdmi-Raise-the-maximum-clock-rate.patch
index d1deaa884128..99fd4f4853f9 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0576-drm-vc4-hdmi-Raise-the-maximum-clock-rate.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0576-drm-vc4-hdmi-Raise-the-maximum-clock-rate.patch
@@ -19,7 +19,7 @@ Signed-off-by: Maxime Ripard <maxime@cerno.tech>
 
 --- a/drivers/gpu/drm/vc4/vc4_hdmi.c
 +++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
-@@ -2432,7 +2432,7 @@ static const struct vc4_hdmi_variant bcm
+@@ -2434,7 +2434,7 @@ static const struct vc4_hdmi_variant bcm
  	.encoder_type		= VC4_ENCODER_TYPE_HDMI0,
  	.debugfs_name		= "hdmi0_regs",
  	.card_name		= "vc4-hdmi-0",
diff --git a/target/linux/bcm27xx/patches-5.10/950-0592-drm-vc4-Allow-DBLCLK-modes-even-if-horz-timing-is-od.patch b/target/linux/bcm27xx/patches-5.10/950-0592-drm-vc4-Allow-DBLCLK-modes-even-if-horz-timing-is-od.patch
deleted file mode 100644
index 459f228c1566..000000000000
--- a/target/linux/bcm27xx/patches-5.10/950-0592-drm-vc4-Allow-DBLCLK-modes-even-if-horz-timing-is-od.patch
+++ /dev/null
@@ -1,42 +0,0 @@
-From 9f7c0728efb0036f6f197126aa62da40cdf4713a Mon Sep 17 00:00:00 2001
-From: Dave Stevenson <dave.stevenson@raspberrypi.com>
-Date: Wed, 28 Apr 2021 16:14:21 +0100
-Subject: [PATCH] drm/vc4: Allow DBLCLK modes even if horz timing is
- odd.
-
-The 2711 pixel valve can't produce odd horizontal timings, and
-checks were added to vc4_hdmi_encoder_atomic_check and
-vc4_hdmi_encoder_mode_valid to filter out/block selection of
-such modes.
-
-Modes with DRM_MODE_FLAG_DBLCLK double all the horizontal timing
-values before programming them into the PV. The PV values,
-therefore, can not be odd, and so the modes can be supported.
-
-Amend the filtering appropriately.
-
-See https://github.com/raspberrypi/linux/issues/4307
-
-Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
----
- drivers/gpu/drm/vc4/vc4_hdmi.c | 2 ++
- 1 file changed, 2 insertions(+)
-
---- a/drivers/gpu/drm/vc4/vc4_hdmi.c
-+++ b/drivers/gpu/drm/vc4/vc4_hdmi.c
-@@ -1067,6 +1067,7 @@ static int vc4_hdmi_encoder_atomic_check
- 	unsigned long long tmds_rate;
- 
- 	if (vc4_hdmi->variant->unsupported_odd_h_timings &&
-+	    !(mode->flags & DRM_MODE_FLAG_DBLCLK) &&
- 	    ((mode->hdisplay % 2) || (mode->hsync_start % 2) ||
- 	     (mode->hsync_end % 2) || (mode->htotal % 2)))
- 		return -EINVAL;
-@@ -1111,6 +1112,7 @@ vc4_hdmi_encoder_mode_valid(struct drm_e
- 	struct vc4_hdmi *vc4_hdmi = encoder_to_vc4_hdmi(encoder);
- 
- 	if (vc4_hdmi->variant->unsupported_odd_h_timings &&
-+	    !(mode->flags & DRM_MODE_FLAG_DBLCLK) &&
- 	    ((mode->hdisplay % 2) || (mode->hsync_start % 2) ||
- 	     (mode->hsync_end % 2) || (mode->htotal % 2)))
- 		return MODE_H_ILLEGAL;
diff --git a/target/linux/bcm27xx/patches-5.10/950-0650-Support-RPi-DPI-interface-in-mode6-for-18-bit-color.patch b/target/linux/bcm27xx/patches-5.10/950-0650-Support-RPi-DPI-interface-in-mode6-for-18-bit-color.patch
index 56a1e6862b03..a5fe888565d4 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0650-Support-RPi-DPI-interface-in-mode6-for-18-bit-color.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0650-Support-RPi-DPI-interface-in-mode6-for-18-bit-color.patch
@@ -118,7 +118,7 @@ Signed-off-by: Joerg Quinten <aBUGSworstnightmare@gmail.com>
        - MEDIA_BUS_FMT_RGB666_1X24_CPADHI
 --- a/drivers/gpu/drm/panel/panel-simple.c
 +++ b/drivers/gpu/drm/panel/panel-simple.c
-@@ -2093,6 +2093,38 @@ static const struct panel_desc innolux_a
+@@ -2094,6 +2094,38 @@ static const struct panel_desc innolux_a
  	.bus_flags = DRM_BUS_FLAG_DE_HIGH | DRM_BUS_FLAG_PIXDATA_DRIVE_POSEDGE,
  };
  
@@ -157,7 +157,7 @@ Signed-off-by: Joerg Quinten <aBUGSworstnightmare@gmail.com>
  static const struct drm_display_mode innolux_at070tn92_mode = {
  	.clock = 33333,
  	.hdisplay = 800,
-@@ -4076,6 +4108,9 @@ static const struct of_device_id platfor
+@@ -4077,6 +4109,9 @@ static const struct of_device_id platfor
  		.compatible = "innolux,at043tn24",
  		.data = &innolux_at043tn24,
  	}, {
diff --git a/target/linux/bcm27xx/patches-5.10/950-0738-Makefiles-dt-Always-set-on-ARCH_BCM2835.patch b/target/linux/bcm27xx/patches-5.10/950-0738-Makefiles-dt-Always-set-on-ARCH_BCM2835.patch
index 0cb30474b591..db966ef86fbf 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0738-Makefiles-dt-Always-set-on-ARCH_BCM2835.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0738-Makefiles-dt-Always-set-on-ARCH_BCM2835.patch
@@ -17,7 +17,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 
 --- a/arch/arm/boot/dts/Makefile
 +++ b/arch/arm/boot/dts/Makefile
-@@ -1432,5 +1432,5 @@ subdir-y	:= overlays
+@@ -1433,5 +1433,5 @@ subdir-y	:= overlays
  
  # Enable fixups to support overlays on BCM2835 platforms
  ifeq ($(CONFIG_ARCH_BCM2835),y)
diff --git a/target/linux/bcm4908/patches-5.10/071-v5.12-0001-net-dsa-bcm_sf2-support-BCM4908-s-integrated-switch.patch b/target/linux/bcm4908/patches-5.10/071-v5.12-0001-net-dsa-bcm_sf2-support-BCM4908-s-integrated-switch.patch
index e2f63e7fa095..35017bf1e434 100644
--- a/target/linux/bcm4908/patches-5.10/071-v5.12-0001-net-dsa-bcm_sf2-support-BCM4908-s-integrated-switch.patch
+++ b/target/linux/bcm4908/patches-5.10/071-v5.12-0001-net-dsa-bcm_sf2-support-BCM4908-s-integrated-switch.patch
@@ -68,7 +68,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  			offset = CORE_STS_OVERRIDE_IMP;
  		else
  			offset = CORE_STS_OVERRIDE_IMP2;
-@@ -708,7 +709,8 @@ static void bcm_sf2_sw_mac_link_down(str
+@@ -711,7 +712,8 @@ static void bcm_sf2_sw_mac_link_down(str
  	u32 reg, offset;
  
  	if (port != core_readl(priv, CORE_IMP0_PRT_ID)) {
@@ -78,7 +78,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  			offset = CORE_STS_OVERRIDE_GMIIP_PORT(port);
  		else
  			offset = CORE_STS_OVERRIDE_GMIIP2_PORT(port);
-@@ -735,7 +737,8 @@ static void bcm_sf2_sw_mac_link_up(struc
+@@ -738,7 +740,8 @@ static void bcm_sf2_sw_mac_link_up(struc
  	bcm_sf2_sw_mac_link_set(ds, port, interface, true);
  
  	if (port != core_readl(priv, CORE_IMP0_PRT_ID)) {
@@ -88,7 +88,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  			offset = CORE_STS_OVERRIDE_GMIIP_PORT(port);
  		else
  			offset = CORE_STS_OVERRIDE_GMIIP2_PORT(port);
-@@ -1128,6 +1131,30 @@ struct bcm_sf2_of_data {
+@@ -1131,6 +1134,30 @@ struct bcm_sf2_of_data {
  	unsigned int num_cfp_rules;
  };
  
@@ -119,7 +119,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  /* Register offsets for the SWITCH_REG_* block */
  static const u16 bcm_sf2_7445_reg_offsets[] = {
  	[REG_SWITCH_CNTRL]	= 0x00,
-@@ -1176,6 +1203,9 @@ static const struct bcm_sf2_of_data bcm_
+@@ -1179,6 +1206,9 @@ static const struct bcm_sf2_of_data bcm_
  };
  
  static const struct of_device_id bcm_sf2_of_match[] = {
diff --git a/target/linux/bcm4908/patches-5.10/075-v5.13-0002-net-dsa-bcm_sf2-setup-BCM4908-internal-crossbar.patch b/target/linux/bcm4908/patches-5.10/075-v5.13-0002-net-dsa-bcm_sf2-setup-BCM4908-internal-crossbar.patch
index c6c0c23570f0..b9ec7af1ae4c 100644
--- a/target/linux/bcm4908/patches-5.10/075-v5.13-0002-net-dsa-bcm_sf2-setup-BCM4908-internal-crossbar.patch
+++ b/target/linux/bcm4908/patches-5.10/075-v5.13-0002-net-dsa-bcm_sf2-setup-BCM4908-internal-crossbar.patch
@@ -82,7 +82,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  static void bcm_sf2_intr_disable(struct bcm_sf2_priv *priv)
  {
  	intrl2_0_mask_set(priv, 0xffffffff);
-@@ -869,6 +907,8 @@ static int bcm_sf2_sw_resume(struct dsa_
+@@ -872,6 +910,8 @@ static int bcm_sf2_sw_resume(struct dsa_
  		return ret;
  	}
  
@@ -91,7 +91,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	ret = bcm_sf2_cfp_resume(ds);
  	if (ret)
  		return ret;
-@@ -1140,6 +1180,7 @@ struct bcm_sf2_of_data {
+@@ -1143,6 +1183,7 @@ struct bcm_sf2_of_data {
  	const u16 *reg_offsets;
  	unsigned int core_reg_align;
  	unsigned int num_cfp_rules;
@@ -99,7 +99,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  };
  
  static const u16 bcm_sf2_4908_reg_offsets[] = {
-@@ -1164,6 +1205,7 @@ static const struct bcm_sf2_of_data bcm_
+@@ -1167,6 +1208,7 @@ static const struct bcm_sf2_of_data bcm_
  	.core_reg_align	= 0,
  	.reg_offsets	= bcm_sf2_4908_reg_offsets,
  	.num_cfp_rules	= 0, /* FIXME */
@@ -107,7 +107,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  };
  
  /* Register offsets for the SWITCH_REG_* block */
-@@ -1274,6 +1316,7 @@ static int bcm_sf2_sw_probe(struct platf
+@@ -1277,6 +1319,7 @@ static int bcm_sf2_sw_probe(struct platf
  	priv->reg_offsets = data->reg_offsets;
  	priv->core_reg_align = data->core_reg_align;
  	priv->num_cfp_rules = data->num_cfp_rules;
@@ -115,7 +115,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  
  	priv->rcdev = devm_reset_control_get_optional_exclusive(&pdev->dev,
  								"switch");
-@@ -1347,6 +1390,8 @@ static int bcm_sf2_sw_probe(struct platf
+@@ -1350,6 +1393,8 @@ static int bcm_sf2_sw_probe(struct platf
  		goto out_clk_mdiv;
  	}
  
diff --git a/target/linux/bcm4908/patches-5.10/075-v5.13-0003-net-dsa-bcm_sf2-Fill-in-BCM4908-CFP-entries.patch b/target/linux/bcm4908/patches-5.10/075-v5.13-0003-net-dsa-bcm_sf2-Fill-in-BCM4908-CFP-entries.patch
index 58fcec27ea4e..b99bf010bd65 100644
--- a/target/linux/bcm4908/patches-5.10/075-v5.13-0003-net-dsa-bcm_sf2-Fill-in-BCM4908-CFP-entries.patch
+++ b/target/linux/bcm4908/patches-5.10/075-v5.13-0003-net-dsa-bcm_sf2-Fill-in-BCM4908-CFP-entries.patch
@@ -14,7 +14,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
 
 --- a/drivers/net/dsa/bcm_sf2.c
 +++ b/drivers/net/dsa/bcm_sf2.c
-@@ -1204,7 +1204,7 @@ static const struct bcm_sf2_of_data bcm_
+@@ -1207,7 +1207,7 @@ static const struct bcm_sf2_of_data bcm_
  	.type		= BCM4908_DEVICE_ID,
  	.core_reg_align	= 0,
  	.reg_offsets	= bcm_sf2_4908_reg_offsets,
diff --git a/target/linux/bcm4908/patches-5.10/075-v5.13-0004-net-dsa-bcm_sf2-add-function-finding-RGMII-register.patch b/target/linux/bcm4908/patches-5.10/075-v5.13-0004-net-dsa-bcm_sf2-add-function-finding-RGMII-register.patch
index 52387776b58b..132a734ac4ef 100644
--- a/target/linux/bcm4908/patches-5.10/075-v5.13-0004-net-dsa-bcm_sf2-add-function-finding-RGMII-register.patch
+++ b/target/linux/bcm4908/patches-5.10/075-v5.13-0004-net-dsa-bcm_sf2-add-function-finding-RGMII-register.patch
@@ -57,7 +57,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  static void bcm_sf2_imp_setup(struct dsa_switch *ds, int port)
  {
  	struct bcm_sf2_priv *priv = bcm_sf2_to_priv(ds);
-@@ -693,6 +718,7 @@ static void bcm_sf2_sw_mac_config(struct
+@@ -696,6 +721,7 @@ static void bcm_sf2_sw_mac_config(struct
  {
  	struct bcm_sf2_priv *priv = bcm_sf2_to_priv(ds);
  	u32 id_mode_dis = 0, port_mode;
@@ -65,7 +65,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	u32 reg;
  
  	if (port == core_readl(priv, CORE_IMP0_PRT_ID))
-@@ -716,10 +742,12 @@ static void bcm_sf2_sw_mac_config(struct
+@@ -719,10 +745,12 @@ static void bcm_sf2_sw_mac_config(struct
  		return;
  	}
  
@@ -79,7 +79,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	reg &= ~ID_MODE_DIS;
  	reg &= ~(PORT_MODE_MASK << PORT_MODE_SHIFT);
  
-@@ -727,13 +755,14 @@ static void bcm_sf2_sw_mac_config(struct
+@@ -730,13 +758,14 @@ static void bcm_sf2_sw_mac_config(struct
  	if (id_mode_dis)
  		reg |= ID_MODE_DIS;
  
@@ -95,7 +95,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	u32 reg;
  
  	if (!phy_interface_mode_is_rgmii(interface) &&
-@@ -741,13 +770,15 @@ static void bcm_sf2_sw_mac_link_set(stru
+@@ -744,13 +773,15 @@ static void bcm_sf2_sw_mac_link_set(stru
  	    interface != PHY_INTERFACE_MODE_REVMII)
  		return;
  
@@ -113,7 +113,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  }
  
  static void bcm_sf2_sw_mac_link_down(struct dsa_switch *ds, int port,
-@@ -781,11 +812,15 @@ static void bcm_sf2_sw_mac_link_up(struc
+@@ -784,11 +815,15 @@ static void bcm_sf2_sw_mac_link_up(struc
  {
  	struct bcm_sf2_priv *priv = bcm_sf2_to_priv(ds);
  	struct ethtool_eee *p = &priv->dev->ports[port].eee;
@@ -130,7 +130,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		if (priv->type == BCM4908_DEVICE_ID ||
  		    priv->type == BCM7445_DEVICE_ID)
  			offset = CORE_STS_OVERRIDE_GMIIP_PORT(port);
-@@ -796,7 +831,7 @@ static void bcm_sf2_sw_mac_link_up(struc
+@@ -799,7 +834,7 @@ static void bcm_sf2_sw_mac_link_up(struc
  		    interface == PHY_INTERFACE_MODE_RGMII_TXID ||
  		    interface == PHY_INTERFACE_MODE_MII ||
  		    interface == PHY_INTERFACE_MODE_REVMII) {
@@ -139,7 +139,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  			reg &= ~(RX_PAUSE_EN | TX_PAUSE_EN);
  
  			if (tx_pause)
-@@ -804,7 +839,7 @@ static void bcm_sf2_sw_mac_link_up(struc
+@@ -807,7 +842,7 @@ static void bcm_sf2_sw_mac_link_up(struc
  			if (rx_pause)
  				reg |= RX_PAUSE_EN;
  
diff --git a/target/linux/bcm4908/patches-5.10/075-v5.13-0005-net-dsa-bcm_sf2-fix-BCM4908-RGMII-reg-s.patch b/target/linux/bcm4908/patches-5.10/075-v5.13-0005-net-dsa-bcm_sf2-fix-BCM4908-RGMII-reg-s.patch
index 1766f5f432a5..67c633a930bb 100644
--- a/target/linux/bcm4908/patches-5.10/075-v5.13-0005-net-dsa-bcm_sf2-fix-BCM4908-RGMII-reg-s.patch
+++ b/target/linux/bcm4908/patches-5.10/075-v5.13-0005-net-dsa-bcm_sf2-fix-BCM4908-RGMII-reg-s.patch
@@ -33,7 +33,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		break;
  	default:
  		switch (port) {
-@@ -1227,9 +1232,7 @@ static const u16 bcm_sf2_4908_reg_offset
+@@ -1230,9 +1235,7 @@ static const u16 bcm_sf2_4908_reg_offset
  	[REG_PHY_REVISION]	= 0x14,
  	[REG_SPHY_CNTRL]	= 0x24,
  	[REG_CROSSBAR]		= 0xc8,
diff --git a/target/linux/bcm4908/patches-5.10/075-v5.13-0006-net-dsa-bcm_sf2-Fix-bcm_sf2_reg_rgmii_cntrl-call-for.patch b/target/linux/bcm4908/patches-5.10/075-v5.13-0006-net-dsa-bcm_sf2-Fix-bcm_sf2_reg_rgmii_cntrl-call-for.patch
index 7a670d701195..5e5d6d939d89 100644
--- a/target/linux/bcm4908/patches-5.10/075-v5.13-0006-net-dsa-bcm_sf2-Fix-bcm_sf2_reg_rgmii_cntrl-call-for.patch
+++ b/target/linux/bcm4908/patches-5.10/075-v5.13-0006-net-dsa-bcm_sf2-Fix-bcm_sf2_reg_rgmii_cntrl-call-for.patch
@@ -23,7 +23,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
 
 --- a/drivers/net/dsa/bcm_sf2.c
 +++ b/drivers/net/dsa/bcm_sf2.c
-@@ -821,11 +821,9 @@ static void bcm_sf2_sw_mac_link_up(struc
+@@ -824,11 +824,9 @@ static void bcm_sf2_sw_mac_link_up(struc
  	bcm_sf2_sw_mac_link_set(ds, port, interface, true);
  
  	if (port != core_readl(priv, CORE_IMP0_PRT_ID)) {
@@ -36,7 +36,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		if (priv->type == BCM4908_DEVICE_ID ||
  		    priv->type == BCM7445_DEVICE_ID)
  			offset = CORE_STS_OVERRIDE_GMIIP_PORT(port);
-@@ -836,6 +834,7 @@ static void bcm_sf2_sw_mac_link_up(struc
+@@ -839,6 +837,7 @@ static void bcm_sf2_sw_mac_link_up(struc
  		    interface == PHY_INTERFACE_MODE_RGMII_TXID ||
  		    interface == PHY_INTERFACE_MODE_MII ||
  		    interface == PHY_INTERFACE_MODE_REVMII) {
diff --git a/target/linux/bcm4908/patches-5.10/076-v5.17-net-dsa-bcm_sf2-refactor-LED-regs-access.patch b/target/linux/bcm4908/patches-5.10/076-v5.17-net-dsa-bcm_sf2-refactor-LED-regs-access.patch
index dc5aa578cd9f..bf45cdcf24be 100644
--- a/target/linux/bcm4908/patches-5.10/076-v5.17-net-dsa-bcm_sf2-refactor-LED-regs-access.patch
+++ b/target/linux/bcm4908/patches-5.10/076-v5.17-net-dsa-bcm_sf2-refactor-LED-regs-access.patch
@@ -82,7 +82,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  	}
  }
  
-@@ -1232,9 +1269,14 @@ static const u16 bcm_sf2_4908_reg_offset
+@@ -1235,9 +1272,14 @@ static const u16 bcm_sf2_4908_reg_offset
  	[REG_SPHY_CNTRL]	= 0x24,
  	[REG_CROSSBAR]		= 0xc8,
  	[REG_RGMII_11_CNTRL]	= 0x014c,
diff --git a/target/linux/bcm4908/patches-5.10/700-net-dsa-bcm_sf2-enable-GPHY-for-switch-probing.patch b/target/linux/bcm4908/patches-5.10/700-net-dsa-bcm_sf2-enable-GPHY-for-switch-probing.patch
index 5150b074a933..a53fca0e9036 100644
--- a/target/linux/bcm4908/patches-5.10/700-net-dsa-bcm_sf2-enable-GPHY-for-switch-probing.patch
+++ b/target/linux/bcm4908/patches-5.10/700-net-dsa-bcm_sf2-enable-GPHY-for-switch-probing.patch
@@ -29,7 +29,7 @@ Signed-off-by: Rafał Miłecki <rafal@milecki.pl>
 
 --- a/drivers/net/dsa/bcm_sf2.c
 +++ b/drivers/net/dsa/bcm_sf2.c
-@@ -1529,10 +1529,14 @@ static int bcm_sf2_sw_probe(struct platf
+@@ -1532,10 +1532,14 @@ static int bcm_sf2_sw_probe(struct platf
  	rev = reg_readl(priv, REG_PHY_REVISION);
  	priv->hw_params.gphy_rev = rev & PHY_REVISION_MASK;
  
diff --git a/target/linux/bcm4908/patches-5.10/701-net-dsa-bcm_sf2-keep-GPHY-enabled-on-the-BCM4908.patch b/target/linux/bcm4908/patches-5.10/701-net-dsa-bcm_sf2-keep-GPHY-enabled-on-the-BCM4908.patch
index 0358a31f4342..00ffb19f6fcf 100644
--- a/target/linux/bcm4908/patches-5.10/701-net-dsa-bcm_sf2-keep-GPHY-enabled-on-the-BCM4908.patch
+++ b/target/linux/bcm4908/patches-5.10/701-net-dsa-bcm_sf2-keep-GPHY-enabled-on-the-BCM4908.patch
@@ -15,7 +15,7 @@ Signed-off-by: Rafał Miłecki <rafal@milecki.pl>
 
 --- a/drivers/net/dsa/bcm_sf2.c
 +++ b/drivers/net/dsa/bcm_sf2.c
-@@ -1543,6 +1543,12 @@ static int bcm_sf2_sw_probe(struct platf
+@@ -1546,6 +1546,12 @@ static int bcm_sf2_sw_probe(struct platf
  		 priv->hw_params.core_rev >> 8, priv->hw_params.core_rev & 0xff,
  		 priv->irq0, priv->irq1);
  
diff --git a/target/linux/generic/backport-5.10/712-v5.13-net-phy-marvell-refactor-HWMON-OOP-style.patch b/target/linux/generic/backport-5.10/712-v5.13-net-phy-marvell-refactor-HWMON-OOP-style.patch
index 13e241a400a3..278df4631337 100644
--- a/target/linux/generic/backport-5.10/712-v5.13-net-phy-marvell-refactor-HWMON-OOP-style.patch
+++ b/target/linux/generic/backport-5.10/712-v5.13-net-phy-marvell-refactor-HWMON-OOP-style.patch
@@ -18,7 +18,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
 
 --- a/drivers/net/phy/marvell.c
 +++ b/drivers/net/phy/marvell.c
-@@ -2140,6 +2140,19 @@ static int marvell_vct7_cable_test_get_s
+@@ -2141,6 +2141,19 @@ static int marvell_vct7_cable_test_get_s
  }
  
  #ifdef CONFIG_HWMON
@@ -38,7 +38,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  static int m88e1121_get_temp(struct phy_device *phydev, long *temp)
  {
  	int oldpage;
-@@ -2183,75 +2196,6 @@ error:
+@@ -2184,75 +2197,6 @@ error:
  	return phy_restore_page(phydev, oldpage, ret);
  }
  
@@ -114,7 +114,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  static int m88e1510_get_temp(struct phy_device *phydev, long *temp)
  {
  	int ret;
-@@ -2314,92 +2258,6 @@ static int m88e1510_get_temp_alarm(struc
+@@ -2315,92 +2259,6 @@ static int m88e1510_get_temp_alarm(struc
  	return 0;
  }
  
@@ -207,7 +207,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  static int m88e6390_get_temp(struct phy_device *phydev, long *temp)
  {
  	int sum = 0;
-@@ -2458,63 +2316,112 @@ error:
+@@ -2459,63 +2317,112 @@ error:
  	return ret;
  }
  
@@ -343,7 +343,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  };
  
  static int marvell_hwmon_name(struct phy_device *phydev)
-@@ -2537,49 +2444,48 @@ static int marvell_hwmon_name(struct phy
+@@ -2538,49 +2445,48 @@ static int marvell_hwmon_name(struct phy
  	return 0;
  }
  
@@ -416,7 +416,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  {
  	return 0;
  }
-@@ -2595,40 +2501,7 @@ static int marvell_probe(struct phy_devi
+@@ -2596,40 +2502,7 @@ static int marvell_probe(struct phy_devi
  
  	phydev->priv = priv;
  
@@ -458,7 +458,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  }
  
  static struct phy_driver marvell_drivers[] = {
-@@ -2713,8 +2586,9 @@ static struct phy_driver marvell_drivers
+@@ -2714,8 +2587,9 @@ static struct phy_driver marvell_drivers
  		.phy_id = MARVELL_PHY_ID_88E1121R,
  		.phy_id_mask = MARVELL_PHY_ID_MASK,
  		.name = "Marvell 88E1121R",
@@ -469,7 +469,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		.config_init = marvell_config_init,
  		.config_aneg = m88e1121_config_aneg,
  		.read_status = marvell_read_status,
-@@ -2833,9 +2707,10 @@ static struct phy_driver marvell_drivers
+@@ -2834,9 +2708,10 @@ static struct phy_driver marvell_drivers
  		.phy_id = MARVELL_PHY_ID_88E1510,
  		.phy_id_mask = MARVELL_PHY_ID_MASK,
  		.name = "Marvell 88E1510",
@@ -481,7 +481,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		.config_init = m88e1510_config_init,
  		.config_aneg = m88e1510_config_aneg,
  		.read_status = marvell_read_status,
-@@ -2862,9 +2737,10 @@ static struct phy_driver marvell_drivers
+@@ -2863,9 +2738,10 @@ static struct phy_driver marvell_drivers
  		.phy_id = MARVELL_PHY_ID_88E1540,
  		.phy_id_mask = MARVELL_PHY_ID_MASK,
  		.name = "Marvell 88E1540",
@@ -493,7 +493,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		.config_init = marvell_config_init,
  		.config_aneg = m88e1510_config_aneg,
  		.read_status = marvell_read_status,
-@@ -2888,7 +2764,8 @@ static struct phy_driver marvell_drivers
+@@ -2889,7 +2765,8 @@ static struct phy_driver marvell_drivers
  		.phy_id = MARVELL_PHY_ID_88E1545,
  		.phy_id_mask = MARVELL_PHY_ID_MASK,
  		.name = "Marvell 88E1545",
@@ -503,7 +503,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		/* PHY_GBIT_FEATURES */
  		.flags = PHY_POLL_CABLE_TEST,
  		.config_init = marvell_config_init,
-@@ -2934,9 +2811,10 @@ static struct phy_driver marvell_drivers
+@@ -2935,9 +2812,10 @@ static struct phy_driver marvell_drivers
  		.phy_id = MARVELL_PHY_ID_88E6341_FAMILY,
  		.phy_id_mask = MARVELL_PHY_ID_MASK,
  		.name = "Marvell 88E6341 Family",
@@ -515,7 +515,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		.config_init = marvell_config_init,
  		.config_aneg = m88e6390_config_aneg,
  		.read_status = marvell_read_status,
-@@ -2960,9 +2838,10 @@ static struct phy_driver marvell_drivers
+@@ -2961,9 +2839,10 @@ static struct phy_driver marvell_drivers
  		.phy_id = MARVELL_PHY_ID_88E6390_FAMILY,
  		.phy_id_mask = MARVELL_PHY_ID_MASK,
  		.name = "Marvell 88E6390 Family",
@@ -527,7 +527,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		.config_init = marvell_config_init,
  		.config_aneg = m88e6390_config_aneg,
  		.read_status = marvell_read_status,
-@@ -2986,7 +2865,8 @@ static struct phy_driver marvell_drivers
+@@ -2987,7 +2866,8 @@ static struct phy_driver marvell_drivers
  		.phy_id = MARVELL_PHY_ID_88E1340S,
  		.phy_id_mask = MARVELL_PHY_ID_MASK,
  		.name = "Marvell 88E1340S",
@@ -537,7 +537,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		/* PHY_GBIT_FEATURES */
  		.config_init = marvell_config_init,
  		.config_aneg = m88e1510_config_aneg,
-@@ -3008,7 +2888,8 @@ static struct phy_driver marvell_drivers
+@@ -3009,7 +2889,8 @@ static struct phy_driver marvell_drivers
  		.phy_id = MARVELL_PHY_ID_88E1548P,
  		.phy_id_mask = MARVELL_PHY_ID_MASK,
  		.name = "Marvell 88E1548P",
diff --git a/target/linux/generic/backport-5.10/713-v5.15-net-phy-marvell-add-SFP-support-for-88E1510.patch b/target/linux/generic/backport-5.10/713-v5.15-net-phy-marvell-add-SFP-support-for-88E1510.patch
index 3e01e727e1c2..b86e9bf64099 100644
--- a/target/linux/generic/backport-5.10/713-v5.15-net-phy-marvell-add-SFP-support-for-88E1510.patch
+++ b/target/linux/generic/backport-5.10/713-v5.15-net-phy-marvell-add-SFP-support-for-88E1510.patch
@@ -49,7 +49,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  #define MII_88E1510_GEN_CTRL_REG_1_RESET	0x8000	/* Soft reset */
  
  #define MII_VCT5_TX_RX_MDI0_COUPLING	0x10
-@@ -2504,6 +2513,100 @@ static int marvell_probe(struct phy_devi
+@@ -2505,6 +2514,100 @@ static int marvell_probe(struct phy_devi
  	return marvell_hwmon_probe(phydev);
  }
  
@@ -150,7 +150,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  static struct phy_driver marvell_drivers[] = {
  	{
  		.phy_id = MARVELL_PHY_ID_88E1101,
-@@ -2710,7 +2813,7 @@ static struct phy_driver marvell_drivers
+@@ -2711,7 +2814,7 @@ static struct phy_driver marvell_drivers
  		.driver_data = DEF_MARVELL_HWMON_OPS(m88e1510_hwmon_ops),
  		.features = PHY_GBIT_FIBRE_FEATURES,
  		.flags = PHY_POLL_CABLE_TEST,
diff --git a/target/linux/generic/pending-5.10/768-net-dsa-mv88e6xxx-Request-assisted-learning-on-CPU-port.patch b/target/linux/generic/pending-5.10/768-net-dsa-mv88e6xxx-Request-assisted-learning-on-CPU-port.patch
index 99f42ff2128c..9247401140cb 100644
--- a/target/linux/generic/pending-5.10/768-net-dsa-mv88e6xxx-Request-assisted-learning-on-CPU-port.patch
+++ b/target/linux/generic/pending-5.10/768-net-dsa-mv88e6xxx-Request-assisted-learning-on-CPU-port.patch
@@ -17,7 +17,7 @@ Signed-off-by: Tobias Waldekranz <tobias@waldekranz.com>
 
 --- a/drivers/net/dsa/mv88e6xxx/chip.c
 +++ b/drivers/net/dsa/mv88e6xxx/chip.c
-@@ -5431,6 +5431,7 @@ static int mv88e6xxx_register_switch(str
+@@ -5436,6 +5436,7 @@ static int mv88e6xxx_register_switch(str
  	ds->ops = &mv88e6xxx_switch_ops;
  	ds->ageing_time_min = chip->info->age_time_coeff;
  	ds->ageing_time_max = chip->info->age_time_coeff * U8_MAX;
diff --git a/target/linux/generic/pending-5.10/841-USB-serial-option-add-ZTE-MF286D-modem.patch b/target/linux/generic/pending-5.10/841-USB-serial-option-add-ZTE-MF286D-modem.patch
deleted file mode 100644
index bdd2c31b416f..000000000000
--- a/target/linux/generic/pending-5.10/841-USB-serial-option-add-ZTE-MF286D-modem.patch
+++ /dev/null
@@ -1,57 +0,0 @@
-From 67ae6887854e4b6e736c34d4fd7087540ed31d4e Mon Sep 17 00:00:00 2001
-From: Pawel Dembicki <paweldembicki@gmail.com>
-Date: Tue, 11 Jan 2022 21:59:50 +0100
-Subject: [PATCH] USB: serial: option: add ZTE MF286D modem
-
-Modem from ZTE MF286D is an Qualcomm MDM9250 based 3G/4G modem.
-
-T:  Bus=02 Lev=01 Prnt=01 Port=00 Cnt=01 Dev#=  3 Spd=5000 MxCh= 0
-D:  Ver= 3.00 Cls=00(>ifc ) Sub=00 Prot=00 MxPS= 9 #Cfgs=  1
-P:  Vendor=19d2 ProdID=1485 Rev=52.87
-S:  Manufacturer=ZTE,Incorporated
-S:  Product=ZTE Technologies MSM
-S:  SerialNumber=MF286DZTED000000
-C:* #Ifs= 7 Cfg#= 1 Atr=80 MxPwr=896mA
-A:  FirstIf#= 0 IfCount= 2 Cls=02(comm.) Sub=06 Prot=00
-I:* If#= 0 Alt= 0 #EPs= 1 Cls=02(comm.) Sub=02 Prot=ff Driver=rndis_host
-E:  Ad=82(I) Atr=03(Int.) MxPS=   8 Ivl=32ms
-I:* If#= 1 Alt= 0 #EPs= 2 Cls=0a(data ) Sub=00 Prot=00 Driver=rndis_host
-E:  Ad=81(I) Atr=02(Bulk) MxPS=1024 Ivl=0ms
-E:  Ad=01(O) Atr=02(Bulk) MxPS=1024 Ivl=0ms
-I:* If#= 2 Alt= 0 #EPs= 2 Cls=ff(vend.) Sub=ff Prot=ff Driver=option
-E:  Ad=83(I) Atr=02(Bulk) MxPS=1024 Ivl=0ms
-E:  Ad=02(O) Atr=02(Bulk) MxPS=1024 Ivl=0ms
-I:* If#= 3 Alt= 0 #EPs= 3 Cls=ff(vend.) Sub=ff Prot=ff Driver=option
-E:  Ad=85(I) Atr=03(Int.) MxPS=  10 Ivl=32ms
-E:  Ad=84(I) Atr=02(Bulk) MxPS=1024 Ivl=0ms
-E:  Ad=03(O) Atr=02(Bulk) MxPS=1024 Ivl=0ms
-I:* If#= 4 Alt= 0 #EPs= 3 Cls=ff(vend.) Sub=ff Prot=ff Driver=option
-E:  Ad=87(I) Atr=03(Int.) MxPS=  10 Ivl=32ms
-E:  Ad=86(I) Atr=02(Bulk) MxPS=1024 Ivl=0ms
-E:  Ad=04(O) Atr=02(Bulk) MxPS=1024 Ivl=0ms
-I:* If#= 5 Alt= 0 #EPs= 3 Cls=ff(vend.) Sub=ff Prot=ff Driver=qmi_wwan
-E:  Ad=88(I) Atr=03(Int.) MxPS=   8 Ivl=32ms
-E:  Ad=8e(I) Atr=02(Bulk) MxPS=1024 Ivl=0ms
-E:  Ad=0f(O) Atr=02(Bulk) MxPS=1024 Ivl=0ms
-I:* If#= 6 Alt= 0 #EPs= 2 Cls=ff(vend.) Sub=42 Prot=01 Driver=usbfs
-E:  Ad=05(O) Atr=02(Bulk) MxPS=1024 Ivl=0ms
-E:  Ad=89(I) Atr=02(Bulk) MxPS=1024 Ivl=0ms
-
-Signed-off-by: Pawel Dembicki <paweldembicki@gmail.com>
-
-https://lore.kernel.org/linux-usb/20220111221205.14662-1-paweldembicki@gmail.com/
----
- drivers/usb/serial/option.c | 2 ++
- 1 file changed, 2 insertions(+)
-
---- a/drivers/usb/serial/option.c
-+++ b/drivers/usb/serial/option.c
-@@ -1649,6 +1649,8 @@ static const struct usb_device_id option
- 	  .driver_info = RSVD(2) },
- 	{ USB_DEVICE_INTERFACE_CLASS(ZTE_VENDOR_ID, 0x1476, 0xff) },	/* GosunCn ZTE WeLink ME3630 (ECM/NCM mode) */
- 	{ USB_DEVICE_AND_INTERFACE_INFO(ZTE_VENDOR_ID, 0x1481, 0xff, 0x00, 0x00) }, /* ZTE MF871A */
-+	{ USB_DEVICE_AND_INTERFACE_INFO(ZTE_VENDOR_ID, 0x1485, 0xff, 0xff, 0xff),  /* ZTE MF286D */
-+	  .driver_info = RSVD(5) },
- 	{ USB_DEVICE_AND_INTERFACE_INFO(ZTE_VENDOR_ID, 0x1533, 0xff, 0xff, 0xff) },
- 	{ USB_DEVICE_AND_INTERFACE_INFO(ZTE_VENDOR_ID, 0x1534, 0xff, 0xff, 0xff) },
- 	{ USB_DEVICE_AND_INTERFACE_INFO(ZTE_VENDOR_ID, 0x1535, 0xff, 0xff, 0xff) },
diff --git a/target/linux/ipq40xx/patches-5.10/901-arm-boot-add-dts-files.patch b/target/linux/ipq40xx/patches-5.10/901-arm-boot-add-dts-files.patch
index c8229f28d738..2d24d59444b4 100644
--- a/target/linux/ipq40xx/patches-5.10/901-arm-boot-add-dts-files.patch
+++ b/target/linux/ipq40xx/patches-5.10/901-arm-boot-add-dts-files.patch
@@ -10,7 +10,7 @@ Signed-off-by: John Crispin <john@phrozen.org>
 
 --- a/arch/arm/boot/dts/Makefile
 +++ b/arch/arm/boot/dts/Makefile
-@@ -902,11 +902,74 @@ dtb-$(CONFIG_ARCH_QCOM) += \
+@@ -903,11 +903,74 @@ dtb-$(CONFIG_ARCH_QCOM) += \
  	qcom-apq8074-dragonboard.dtb \
  	qcom-apq8084-ifc6540.dtb \
  	qcom-apq8084-mtp.dtb \
diff --git a/target/linux/ipq806x/patches-5.10/0069-arm-boot-add-dts-files.patch b/target/linux/ipq806x/patches-5.10/0069-arm-boot-add-dts-files.patch
index 49ff218040c8..275b960c188f 100644
--- a/target/linux/ipq806x/patches-5.10/0069-arm-boot-add-dts-files.patch
+++ b/target/linux/ipq806x/patches-5.10/0069-arm-boot-add-dts-files.patch
@@ -10,7 +10,7 @@ Signed-off-by: John Crispin <john@phrozen.org>
 
 --- a/arch/arm/boot/dts/Makefile
 +++ b/arch/arm/boot/dts/Makefile
-@@ -907,8 +907,29 @@ dtb-$(CONFIG_ARCH_QCOM) += \
+@@ -908,8 +908,29 @@ dtb-$(CONFIG_ARCH_QCOM) += \
  	qcom-ipq4019-ap.dk04.1-c3.dtb \
  	qcom-ipq4019-ap.dk07.1-c1.dtb \
  	qcom-ipq4019-ap.dk07.1-c2.dtb \
diff --git a/target/linux/lantiq/patches-5.10/0703-net-lantiq-enable-jumbo-frames-on-GSWIP.patch b/target/linux/lantiq/patches-5.10/0703-net-lantiq-enable-jumbo-frames-on-GSWIP.patch
index 9ff0baed4320..6593e43db709 100644
--- a/target/linux/lantiq/patches-5.10/0703-net-lantiq-enable-jumbo-frames-on-GSWIP.patch
+++ b/target/linux/lantiq/patches-5.10/0703-net-lantiq-enable-jumbo-frames-on-GSWIP.patch
@@ -30,7 +30,7 @@ Signed-off-by: Thomas Nixon <tom@tomn.co.uk>
  struct gswip_hw_info {
  	int max_ports;
  	int cpu_port;
-@@ -851,10 +856,6 @@ static int gswip_setup(struct dsa_switch
+@@ -856,10 +861,6 @@ static int gswip_setup(struct dsa_switch
  	gswip_switch_mask(priv, 0, GSWIP_PCE_PCTRL_0_INGRESS,
  			  GSWIP_PCE_PCTRL_0p(cpu_port));
  
@@ -41,7 +41,7 @@ Signed-off-by: Thomas Nixon <tom@tomn.co.uk>
  	gswip_switch_mask(priv, 0, GSWIP_BM_QUEUE_GCTRL_GL_MOD,
  			  GSWIP_BM_QUEUE_GCTRL);
  
-@@ -871,6 +872,8 @@ static int gswip_setup(struct dsa_switch
+@@ -876,6 +877,8 @@ static int gswip_setup(struct dsa_switch
  		return err;
  	}
  
@@ -50,7 +50,7 @@ Signed-off-by: Thomas Nixon <tom@tomn.co.uk>
  	gswip_port_enable(ds, cpu_port, NULL);
  	return 0;
  }
-@@ -1433,6 +1436,39 @@ static int gswip_port_fdb_dump(struct ds
+@@ -1438,6 +1441,39 @@ static int gswip_port_fdb_dump(struct ds
  	return 0;
  }
  
@@ -90,7 +90,7 @@ Signed-off-by: Thomas Nixon <tom@tomn.co.uk>
  static void gswip_phylink_validate(struct dsa_switch *ds, int port,
  				   unsigned long *supported,
  				   struct phylink_link_state *state)
-@@ -1776,6 +1812,8 @@ static const struct dsa_switch_ops gswip
+@@ -1781,6 +1817,8 @@ static const struct dsa_switch_ops gswip
  	.port_fdb_add		= gswip_port_fdb_add,
  	.port_fdb_del		= gswip_port_fdb_del,
  	.port_fdb_dump		= gswip_port_fdb_dump,
diff --git a/target/linux/sunxi/patches-5.10/062-add-sun8i-h3-zeropi-support.patch b/target/linux/sunxi/patches-5.10/062-add-sun8i-h3-zeropi-support.patch
index 893662b8e523..35e14aa4efce 100644
--- a/target/linux/sunxi/patches-5.10/062-add-sun8i-h3-zeropi-support.patch
+++ b/target/linux/sunxi/patches-5.10/062-add-sun8i-h3-zeropi-support.patch
@@ -1,6 +1,6 @@
 --- a/arch/arm/boot/dts/Makefile
 +++ b/arch/arm/boot/dts/Makefile
-@@ -1202,6 +1202,7 @@ dtb-$(CONFIG_MACH_SUN8I) += \
+@@ -1203,6 +1203,7 @@ dtb-$(CONFIG_MACH_SUN8I) += \
  	sun8i-h3-orangepi-zero-plus2.dtb \
  	sun8i-h3-rervision-dvk.dtb \
  	sun8i-h3-emlid-neutis-n5h3-devboard.dtb \
diff --git a/target/linux/sunxi/patches-5.10/100-sunxi-h3-add-support-for-nanopi-r1.patch b/target/linux/sunxi/patches-5.10/100-sunxi-h3-add-support-for-nanopi-r1.patch
index 5a80fb5f006d..9a969d5e00c4 100644
--- a/target/linux/sunxi/patches-5.10/100-sunxi-h3-add-support-for-nanopi-r1.patch
+++ b/target/linux/sunxi/patches-5.10/100-sunxi-h3-add-support-for-nanopi-r1.patch
@@ -27,7 +27,7 @@ Signed-off-by: Jayantajit Gogoi <jayanta.gogoi525@gmail.com>
            - const: gemei,g9
 --- a/arch/arm/boot/dts/Makefile
 +++ b/arch/arm/boot/dts/Makefile
-@@ -1192,6 +1192,7 @@ dtb-$(CONFIG_MACH_SUN8I) += \
+@@ -1193,6 +1193,7 @@ dtb-$(CONFIG_MACH_SUN8I) += \
  	sun8i-h3-nanopi-m1-plus.dtb \
  	sun8i-h3-nanopi-neo.dtb \
  	sun8i-h3-nanopi-neo-air.dtb \

From 06d0617ba6511f7a46cdfe1ccd3eab7dd8f3a6e4 Mon Sep 17 00:00:00 2001
From: John Audia <graysky@archlinux.us>
Date: Wed, 23 Feb 2022 06:58:46 -0500
Subject: [PATCH 2/2] kernel: bump 5.10 to 5.10.102

Removed upstreamed:
	bcm4908/patches-5.10/180-i2c-brcmstb-fix-support-for-DSL-and-CM-variants.patch[1]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.102&id=f333c1916fd6b55900029bf8f918cc00009e2111

Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200

Signed-off-by: John Audia <graysky@archlinux.us>
---
 include/kernel-5.10                           |  4 +-
 .../910-unaligned_access_hacks.patch          | 10 ++---
 .../patches-5.10/920-mikrotik-rb4xx.patch     |  2 +-
 .../patches-5.10/939-mikrotik-rb91x.patch     |  4 +-
 ...ce-quirks-for-Freeway-Airmouse-T3-an.patch |  2 +-
 .../950-0348-gpio-Add-gpio-fsm-driver.patch   |  2 +-
 ...usb-improve-getting-OF-matching-data.patch |  4 +-
 ...-specify-init-function-format-at-str.patch |  6 +--
 ...-brcm-usb-support-PHY-on-the-BCM4908.patch |  2 +-
 ...-fix-support-for-DSL-and-CM-variants.patch | 30 ---------------
 ...ng-with-source-address-failed-policy.patch |  2 +-
 ...d-knob-for-filtering-rx-tx-BPDU-pack.patch |  2 +-
 ...pio-cascade-add-generic-GPIO-cascade.patch | 18 ++-------
 .../pending-5.10/834-ledtrig-libata.patch     |  8 ++--
 ...m_nandc-add-boot_layout_mode-support.patch | 38 +++++++++----------
 .../oxnas/patches-5.10/999-libata-hacks.patch |  4 +-
 ...alink-add-gpio-driver-for-ralink-SoC.patch |  2 +-
 ...3-gpio-add-realtek-otto-gpio-support.patch |  2 +-
 .../301-gpio-add-rtl8231-driver.patch         |  2 +-
 19 files changed, 52 insertions(+), 92 deletions(-)
 delete mode 100644 target/linux/bcm4908/patches-5.10/180-i2c-brcmstb-fix-support-for-DSL-and-CM-variants.patch

diff --git a/include/kernel-5.10 b/include/kernel-5.10
index 47c2fd41f5b3..0522a1432ff7 100644
--- a/include/kernel-5.10
+++ b/include/kernel-5.10
@@ -1,2 +1,2 @@
-LINUX_VERSION-5.10 = .101
-LINUX_KERNEL_HASH-5.10.101 = 0749258cb7ee47c478dfc34857e8930d41bfe0b074e80e9f57cbfa56d0ba1c8e
+LINUX_VERSION-5.10 = .102
+LINUX_KERNEL_HASH-5.10.102 = e6414895b1c2e406a41a8744160e2b4ef2095e6cebe5363be460d1b198f0e379
diff --git a/target/linux/ath79/patches-5.10/910-unaligned_access_hacks.patch b/target/linux/ath79/patches-5.10/910-unaligned_access_hacks.patch
index e0bd900479aa..cee1db1545f7 100644
--- a/target/linux/ath79/patches-5.10/910-unaligned_access_hacks.patch
+++ b/target/linux/ath79/patches-5.10/910-unaligned_access_hacks.patch
@@ -579,7 +579,7 @@
  
  #define	IP6_MF		0x0001
  #define	IP6_OFFSET	0xFFF8
-@@ -557,8 +557,8 @@ static inline void __ipv6_addr_set_half(
+@@ -560,8 +560,8 @@ static inline void __ipv6_addr_set_half(
  	}
  #endif
  #endif
@@ -590,7 +590,7 @@
  }
  
  static inline void ipv6_addr_set(struct in6_addr *addr,
-@@ -617,6 +617,8 @@ static inline bool ipv6_prefix_equal(con
+@@ -620,6 +620,8 @@ static inline bool ipv6_prefix_equal(con
  	const __be32 *a1 = addr1->s6_addr32;
  	const __be32 *a2 = addr2->s6_addr32;
  	unsigned int pdw, pbi;
@@ -599,7 +599,7 @@
  
  	/* check complete u32 in prefix */
  	pdw = prefixlen >> 5;
-@@ -625,7 +627,9 @@ static inline bool ipv6_prefix_equal(con
+@@ -628,7 +630,9 @@ static inline bool ipv6_prefix_equal(con
  
  	/* check incomplete u32 in prefix */
  	pbi = prefixlen & 0x1f;
@@ -610,7 +610,7 @@
  		return false;
  
  	return true;
-@@ -746,13 +750,13 @@ static inline void ipv6_addr_set_v4mappe
+@@ -749,13 +753,13 @@ static inline void ipv6_addr_set_v4mappe
   */
  static inline int __ipv6_addr_diff32(const void *token1, const void *token2, int addrlen)
  {
@@ -626,7 +626,7 @@
  		if (xb)
  			return i * 32 + 31 - __fls(ntohl(xb));
  	}
-@@ -938,17 +942,18 @@ static inline int ip6_multipath_hash_pol
+@@ -941,17 +945,18 @@ static inline int ip6_multipath_hash_pol
  static inline void ip6_flow_hdr(struct ipv6hdr *hdr, unsigned int tclass,
  				__be32 flowlabel)
  {
diff --git a/target/linux/ath79/patches-5.10/920-mikrotik-rb4xx.patch b/target/linux/ath79/patches-5.10/920-mikrotik-rb4xx.patch
index 3129fac3abac..4ed13deb5c4f 100644
--- a/target/linux/ath79/patches-5.10/920-mikrotik-rb4xx.patch
+++ b/target/linux/ath79/patches-5.10/920-mikrotik-rb4xx.patch
@@ -42,7 +42,7 @@
  menu "SPI GPIO expanders"
 --- a/drivers/gpio/Makefile
 +++ b/drivers/gpio/Makefile
-@@ -120,6 +120,7 @@ obj-$(CONFIG_GPIO_PL061)		+= gpio-pl061.
+@@ -121,6 +121,7 @@ obj-$(CONFIG_GPIO_PL061)		+= gpio-pl061.
  obj-$(CONFIG_GPIO_PMIC_EIC_SPRD)	+= gpio-pmic-eic-sprd.o
  obj-$(CONFIG_GPIO_PXA)			+= gpio-pxa.o
  obj-$(CONFIG_GPIO_RASPBERRYPI_EXP)	+= gpio-raspberrypi-exp.o
diff --git a/target/linux/ath79/patches-5.10/939-mikrotik-rb91x.patch b/target/linux/ath79/patches-5.10/939-mikrotik-rb91x.patch
index 4f4344b40f95..4cf6bb08dfb4 100644
--- a/target/linux/ath79/patches-5.10/939-mikrotik-rb91x.patch
+++ b/target/linux/ath79/patches-5.10/939-mikrotik-rb91x.patch
@@ -27,7 +27,7 @@
  	depends on MFD_SYSCON
 --- a/drivers/gpio/Makefile
 +++ b/drivers/gpio/Makefile
-@@ -72,6 +72,7 @@ obj-$(CONFIG_GPIO_IT87)			+= gpio-it87.o
+@@ -73,6 +73,7 @@ obj-$(CONFIG_GPIO_IT87)			+= gpio-it87.o
  obj-$(CONFIG_GPIO_IXP4XX)		+= gpio-ixp4xx.o
  obj-$(CONFIG_GPIO_JANZ_TTL)		+= gpio-janz-ttl.o
  obj-$(CONFIG_GPIO_KEMPLD)		+= gpio-kempld.o
@@ -35,7 +35,7 @@
  obj-$(CONFIG_GPIO_LOGICVC)		+= gpio-logicvc.o
  obj-$(CONFIG_GPIO_LOONGSON1)		+= gpio-loongson1.o
  obj-$(CONFIG_GPIO_LOONGSON)		+= gpio-loongson.o
-@@ -121,6 +122,7 @@ obj-$(CONFIG_GPIO_PMIC_EIC_SPRD)	+= gpio
+@@ -122,6 +123,7 @@ obj-$(CONFIG_GPIO_PMIC_EIC_SPRD)	+= gpio
  obj-$(CONFIG_GPIO_PXA)			+= gpio-pxa.o
  obj-$(CONFIG_GPIO_RASPBERRYPI_EXP)	+= gpio-raspberrypi-exp.o
  obj-$(CONFIG_GPIO_RB4XX)		+= gpio-rb4xx.o
diff --git a/target/linux/bcm27xx/patches-5.10/950-0151-hid-usb-Add-device-quirks-for-Freeway-Airmouse-T3-an.patch b/target/linux/bcm27xx/patches-5.10/950-0151-hid-usb-Add-device-quirks-for-Freeway-Airmouse-T3-an.patch
index fa76e0a82294..26ced7adfef8 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0151-hid-usb-Add-device-quirks-for-Freeway-Airmouse-T3-an.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0151-hid-usb-Add-device-quirks-for-Freeway-Airmouse-T3-an.patch
@@ -53,7 +53,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.org>
  	{ HID_USB_DEVICE(USB_VENDOR_ID_CHICONY, USB_DEVICE_ID_CHICONY_MULTI_TOUCH), HID_QUIRK_MULTI_INPUT },
  	{ HID_USB_DEVICE(USB_VENDOR_ID_CHICONY, USB_DEVICE_ID_CHICONY_PIXART_USB_OPTICAL_MOUSE), HID_QUIRK_ALWAYS_POLL },
  	{ HID_USB_DEVICE(USB_VENDOR_ID_CHICONY, USB_DEVICE_ID_CHICONY_PIXART_USB_OPTICAL_MOUSE2), HID_QUIRK_ALWAYS_POLL },
-@@ -194,6 +195,7 @@ static const struct hid_device_id hid_qu
+@@ -195,6 +196,7 @@ static const struct hid_device_id hid_qu
  	{ HID_USB_DEVICE(USB_VENDOR_ID_WISEGROUP, USB_DEVICE_ID_QUAD_USB_JOYPAD), HID_QUIRK_NOGET | HID_QUIRK_MULTI_INPUT },
  	{ HID_USB_DEVICE(USB_VENDOR_ID_XIN_MO, USB_DEVICE_ID_XIN_MO_DUAL_ARCADE), HID_QUIRK_MULTI_INPUT },
  	{ HID_USB_DEVICE(USB_VENDOR_ID_LOGITECH, USB_DEVICE_ID_LOGITECH_GROUP_AUDIO), HID_QUIRK_NOGET },
diff --git a/target/linux/bcm27xx/patches-5.10/950-0348-gpio-Add-gpio-fsm-driver.patch b/target/linux/bcm27xx/patches-5.10/950-0348-gpio-Add-gpio-fsm-driver.patch
index 9d2e2f87c65b..2fdef53ea732 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0348-gpio-Add-gpio-fsm-driver.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0348-gpio-Add-gpio-fsm-driver.patch
@@ -42,7 +42,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  	depends on MFD_JANZ_CMODIO
 --- a/drivers/gpio/Makefile
 +++ b/drivers/gpio/Makefile
-@@ -60,6 +60,7 @@ obj-$(CONFIG_GPIO_EP93XX)		+= gpio-ep93x
+@@ -61,6 +61,7 @@ obj-$(CONFIG_GPIO_EP93XX)		+= gpio-ep93x
  obj-$(CONFIG_GPIO_EXAR)			+= gpio-exar.o
  obj-$(CONFIG_GPIO_F7188X)		+= gpio-f7188x.o
  obj-$(CONFIG_GPIO_FTGPIO010)		+= gpio-ftgpio010.o
diff --git a/target/linux/bcm4908/patches-5.10/086-v5.12-0001-phy-phy-brcm-usb-improve-getting-OF-matching-data.patch b/target/linux/bcm4908/patches-5.10/086-v5.12-0001-phy-phy-brcm-usb-improve-getting-OF-matching-data.patch
index cc713731650c..ec9b1fea3d13 100644
--- a/target/linux/bcm4908/patches-5.10/086-v5.12-0001-phy-phy-brcm-usb-improve-getting-OF-matching-data.patch
+++ b/target/linux/bcm4908/patches-5.10/086-v5.12-0001-phy-phy-brcm-usb-improve-getting-OF-matching-data.patch
@@ -27,7 +27,7 @@ Signed-off-by: Vinod Koul <vkoul@kernel.org>
  #include <linux/phy/phy.h>
  #include <linux/platform_device.h>
  #include <linux/interrupt.h>
-@@ -427,7 +428,6 @@ static int brcm_usb_phy_probe(struct pla
+@@ -457,7 +458,6 @@ static int brcm_usb_phy_probe(struct pla
  	struct device_node *dn = pdev->dev.of_node;
  	int err;
  	const char *mode;
@@ -35,7 +35,7 @@ Signed-off-by: Vinod Koul <vkoul@kernel.org>
  	void (*dvr_init)(struct brcm_usb_init_params *params);
  	const struct match_chip_info *info;
  	struct regmap *rmap;
-@@ -441,8 +441,9 @@ static int brcm_usb_phy_probe(struct pla
+@@ -471,8 +471,9 @@ static int brcm_usb_phy_probe(struct pla
  	priv->ini.family_id = brcmstb_get_family_id();
  	priv->ini.product_id = brcmstb_get_product_id();
  
diff --git a/target/linux/bcm4908/patches-5.10/086-v5.12-0002-phy-phy-brcm-usb-specify-init-function-format-at-str.patch b/target/linux/bcm4908/patches-5.10/086-v5.12-0002-phy-phy-brcm-usb-specify-init-function-format-at-str.patch
index bdc932732c69..ffb064f95fd5 100644
--- a/target/linux/bcm4908/patches-5.10/086-v5.12-0002-phy-phy-brcm-usb-specify-init-function-format-at-str.patch
+++ b/target/linux/bcm4908/patches-5.10/086-v5.12-0002-phy-phy-brcm-usb-specify-init-function-format-at-str.patch
@@ -20,7 +20,7 @@ Signed-off-by: Vinod Koul <vkoul@kernel.org>
 
 --- a/drivers/phy/broadcom/phy-brcm-usb.c
 +++ b/drivers/phy/broadcom/phy-brcm-usb.c
-@@ -35,7 +35,7 @@ struct value_to_name_map {
+@@ -36,7 +36,7 @@ struct value_to_name_map {
  };
  
  struct match_chip_info {
@@ -29,7 +29,7 @@ Signed-off-by: Vinod Koul <vkoul@kernel.org>
  	u8 required_regs[BRCM_REGS_MAX + 1];
  	u8 optional_reg;
  };
-@@ -428,7 +428,6 @@ static int brcm_usb_phy_probe(struct pla
+@@ -458,7 +458,6 @@ static int brcm_usb_phy_probe(struct pla
  	struct device_node *dn = pdev->dev.of_node;
  	int err;
  	const char *mode;
@@ -37,7 +37,7 @@ Signed-off-by: Vinod Koul <vkoul@kernel.org>
  	const struct match_chip_info *info;
  	struct regmap *rmap;
  	int x;
-@@ -444,8 +443,8 @@ static int brcm_usb_phy_probe(struct pla
+@@ -474,8 +473,8 @@ static int brcm_usb_phy_probe(struct pla
  	info = of_device_get_match_data(&pdev->dev);
  	if (!info)
  		return -ENOENT;
diff --git a/target/linux/bcm4908/patches-5.10/086-v5.12-0005-phy-phy-brcm-usb-support-PHY-on-the-BCM4908.patch b/target/linux/bcm4908/patches-5.10/086-v5.12-0005-phy-phy-brcm-usb-support-PHY-on-the-BCM4908.patch
index a2cea0f13890..362738c80255 100644
--- a/target/linux/bcm4908/patches-5.10/086-v5.12-0005-phy-phy-brcm-usb-support-PHY-on-the-BCM4908.patch
+++ b/target/linux/bcm4908/patches-5.10/086-v5.12-0005-phy-phy-brcm-usb-support-PHY-on-the-BCM4908.patch
@@ -35,7 +35,7 @@ Signed-off-by: Vinod Koul <vkoul@kernel.org>
  	  Enable this to support the Broadcom STB USB PHY.
 --- a/drivers/phy/broadcom/phy-brcm-usb.c
 +++ b/drivers/phy/broadcom/phy-brcm-usb.c
-@@ -287,6 +287,10 @@ static const struct match_chip_info chip
+@@ -317,6 +317,10 @@ static const struct match_chip_info chip
  
  static const struct of_device_id brcm_usb_dt_ids[] = {
  	{
diff --git a/target/linux/bcm4908/patches-5.10/180-i2c-brcmstb-fix-support-for-DSL-and-CM-variants.patch b/target/linux/bcm4908/patches-5.10/180-i2c-brcmstb-fix-support-for-DSL-and-CM-variants.patch
deleted file mode 100644
index 4efe4fa1cf73..000000000000
--- a/target/linux/bcm4908/patches-5.10/180-i2c-brcmstb-fix-support-for-DSL-and-CM-variants.patch
+++ /dev/null
@@ -1,30 +0,0 @@
-From: =?UTF-8?q?Rafa=C5=82=20Mi=C5=82ecki?= <rafal@milecki.pl>
-Date: Tue, 15 Feb 2022 08:27:35 +0100
-Subject: [PATCH] i2c: brcmstb: fix support for DSL and CM variants
-MIME-Version: 1.0
-Content-Type: text/plain; charset=UTF-8
-Content-Transfer-Encoding: 8bit
-
-DSL and CM (Cable Modem) support 8 B max transfer size and have a custom
-DT binding for that reason. This driver was checking for a wrong
-"compatible" however which resulted in an incorrect setup.
-
-Fixes: e2e5a2c61837 ("i2c: brcmstb: Adding support for CM and DSL SoCs")
-Cc: Kamal Dasu <kdasu.kdev@gmail.com>
-Signed-off-by: Rafał Miłecki <rafal@milecki.pl>
-Acked-by: Florian Fainelli <f.fainelli@gmail.com>
----
- drivers/i2c/busses/i2c-brcmstb.c | 2 +-
- 1 file changed, 1 insertion(+), 1 deletion(-)
-
---- a/drivers/i2c/busses/i2c-brcmstb.c
-+++ b/drivers/i2c/busses/i2c-brcmstb.c
-@@ -674,7 +674,7 @@ static int brcmstb_i2c_probe(struct plat
- 
- 	/* set the data in/out register size for compatible SoCs */
- 	if (of_device_is_compatible(dev->device->of_node,
--				    "brcmstb,brcmper-i2c"))
-+				    "brcm,brcmper-i2c"))
- 		dev->data_regsz = sizeof(u8);
- 	else
- 		dev->data_regsz = sizeof(u32);
diff --git a/target/linux/generic/pending-5.10/670-ipv6-allow-rejecting-with-source-address-failed-policy.patch b/target/linux/generic/pending-5.10/670-ipv6-allow-rejecting-with-source-address-failed-policy.patch
index 68edf37bf79c..d85735531bf2 100644
--- a/target/linux/generic/pending-5.10/670-ipv6-allow-rejecting-with-source-address-failed-policy.patch
+++ b/target/linux/generic/pending-5.10/670-ipv6-allow-rejecting-with-source-address-failed-policy.patch
@@ -20,7 +20,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
 
 --- a/include/net/netns/ipv6.h
 +++ b/include/net/netns/ipv6.h
-@@ -87,6 +87,7 @@ struct netns_ipv6 {
+@@ -88,6 +88,7 @@ struct netns_ipv6 {
  	unsigned int		fib6_routes_require_src;
  #endif
  	struct rt6_info         *ip6_prohibit_entry;
diff --git a/target/linux/generic/pending-5.10/710-bridge-add-knob-for-filtering-rx-tx-BPDU-pack.patch b/target/linux/generic/pending-5.10/710-bridge-add-knob-for-filtering-rx-tx-BPDU-pack.patch
index 0f1df84b306c..ecd6e9e7a250 100644
--- a/target/linux/generic/pending-5.10/710-bridge-add-knob-for-filtering-rx-tx-BPDU-pack.patch
+++ b/target/linux/generic/pending-5.10/710-bridge-add-knob-for-filtering-rx-tx-BPDU-pack.patch
@@ -164,7 +164,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  
  struct rtnl_link {
  	rtnl_doit_func		doit;
-@@ -4682,7 +4682,9 @@ int ndo_dflt_bridge_getlink(struct sk_bu
+@@ -4684,7 +4684,9 @@ int ndo_dflt_bridge_getlink(struct sk_bu
  	    brport_nla_put_flag(skb, flags, mask,
  				IFLA_BRPORT_MCAST_FLOOD, BR_MCAST_FLOOD) ||
  	    brport_nla_put_flag(skb, flags, mask,
diff --git a/target/linux/generic/pending-5.10/801-gpio-gpio-cascade-add-generic-GPIO-cascade.patch b/target/linux/generic/pending-5.10/801-gpio-gpio-cascade-add-generic-GPIO-cascade.patch
index 8d3b3cf9628e..af44ff24aca1 100644
--- a/target/linux/generic/pending-5.10/801-gpio-gpio-cascade-add-generic-GPIO-cascade.patch
+++ b/target/linux/generic/pending-5.10/801-gpio-gpio-cascade-add-generic-GPIO-cascade.patch
@@ -68,13 +68,11 @@ v1 -> v2:
  3 files changed, 133 insertions(+)
  create mode 100644 drivers/gpio/gpio-cascade.c
 
-diff --git a/drivers/gpio/Kconfig b/drivers/gpio/Kconfig
-index 37a6f77c86fe..e69457144459 100644
 --- a/drivers/gpio/Kconfig
 +++ b/drivers/gpio/Kconfig
-@@ -1694,4 +1694,19 @@ config GPIO_VIRTIO
- 
- endmenu
+@@ -1617,4 +1617,19 @@ config GPIO_MOCKUP
+ 	  tools/testing/selftests/gpio/gpio-mockup.sh. Reference the usage in
+ 	  it.
  
 +comment "Other GPIO expanders"
 +
@@ -92,11 +90,9 @@ index 37a6f77c86fe..e69457144459 100644
 +	  will be called 'gpio-cascade'.
 +
  endif
-diff --git a/drivers/gpio/Makefile b/drivers/gpio/Makefile
-index 71ee9fc2ff83..e8945456e7ea 100644
 --- a/drivers/gpio/Makefile
 +++ b/drivers/gpio/Makefile
-@@ -45,6 +45,7 @@ obj-$(CONFIG_GPIO_BD9571MWV)		+= gpio-bd9571mwv.o
+@@ -44,6 +44,7 @@ obj-$(CONFIG_GPIO_BD9571MWV)		+= gpio-bd
  obj-$(CONFIG_GPIO_BRCMSTB)		+= gpio-brcmstb.o
  obj-$(CONFIG_GPIO_BT8XX)		+= gpio-bt8xx.o
  obj-$(CONFIG_GPIO_CADENCE)		+= gpio-cadence.o
@@ -104,9 +100,6 @@ index 71ee9fc2ff83..e8945456e7ea 100644
  obj-$(CONFIG_GPIO_CLPS711X)		+= gpio-clps711x.o
  obj-$(CONFIG_GPIO_SNPS_CREG)		+= gpio-creg-snps.o
  obj-$(CONFIG_GPIO_CRYSTAL_COVE)		+= gpio-crystalcove.o
-diff --git a/drivers/gpio/gpio-cascade.c b/drivers/gpio/gpio-cascade.c
-new file mode 100644
-index 000000000000..5cbda882d79a
 --- /dev/null
 +++ b/drivers/gpio/gpio-cascade.c
 @@ -0,0 +1,117 @@
@@ -227,6 +220,3 @@ index 000000000000..5cbda882d79a
 +MODULE_AUTHOR("Mauri Sandberg <maukka@ext.kapsi.fi>");
 +MODULE_DESCRIPTION("Generic GPIO cascade");
 +MODULE_LICENSE("GPL");
--- 
-2.25.1
-
diff --git a/target/linux/generic/pending-5.10/834-ledtrig-libata.patch b/target/linux/generic/pending-5.10/834-ledtrig-libata.patch
index e43a97bfe8ac..25cffb1444db 100644
--- a/target/linux/generic/pending-5.10/834-ledtrig-libata.patch
+++ b/target/linux/generic/pending-5.10/834-ledtrig-libata.patch
@@ -65,7 +65,7 @@ Signed-off-by: Daniel Golle <daniel@makrotopia.org>
  /**
   *	ata_build_rw_tf - Build ATA taskfile for given read/write request
   *	@tf: Target ATA taskfile
-@@ -4547,6 +4560,9 @@ struct ata_queued_cmd *ata_qc_new_init(s
+@@ -4548,6 +4561,9 @@ struct ata_queued_cmd *ata_qc_new_init(s
  		if (tag < 0)
  			return NULL;
  	}
@@ -75,7 +75,7 @@ Signed-off-by: Daniel Golle <daniel@makrotopia.org>
  
  	qc = __ata_qc_from_tag(ap, tag);
  	qc->tag = qc->hw_tag = tag;
-@@ -5325,6 +5341,9 @@ struct ata_port *ata_port_alloc(struct a
+@@ -5326,6 +5342,9 @@ struct ata_port *ata_port_alloc(struct a
  	ap->stats.unhandled_irq = 1;
  	ap->stats.idle_irq = 1;
  #endif
@@ -85,7 +85,7 @@ Signed-off-by: Daniel Golle <daniel@makrotopia.org>
  	ata_sff_port_init(ap);
  
  	return ap;
-@@ -5360,6 +5379,12 @@ static void ata_host_release(struct kref
+@@ -5361,6 +5380,12 @@ static void ata_host_release(struct kref
  
  		kfree(ap->pmp_link);
  		kfree(ap->slave_link);
@@ -98,7 +98,7 @@ Signed-off-by: Daniel Golle <daniel@makrotopia.org>
  		kfree(ap);
  		host->ports[i] = NULL;
  	}
-@@ -5766,7 +5791,23 @@ int ata_host_register(struct ata_host *h
+@@ -5767,7 +5792,23 @@ int ata_host_register(struct ata_host *h
  		host->ports[i]->print_id = atomic_inc_return(&ata_print_id);
  		host->ports[i]->local_port_no = i + 1;
  	}
diff --git a/target/linux/ipq806x/patches-5.10/099-1-mtd-nand-raw-qcom_nandc-add-boot_layout_mode-support.patch b/target/linux/ipq806x/patches-5.10/099-1-mtd-nand-raw-qcom_nandc-add-boot_layout_mode-support.patch
index 6106dfb8dc12..37501bc64b33 100644
--- a/target/linux/ipq806x/patches-5.10/099-1-mtd-nand-raw-qcom_nandc-add-boot_layout_mode-support.patch
+++ b/target/linux/ipq806x/patches-5.10/099-1-mtd-nand-raw-qcom_nandc-add-boot_layout_mode-support.patch
@@ -16,7 +16,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
 
 --- a/drivers/mtd/nand/raw/qcom_nandc.c
 +++ b/drivers/mtd/nand/raw/qcom_nandc.c
-@@ -159,6 +159,11 @@
+@@ -158,6 +158,11 @@
  /* NAND_CTRL bits */
  #define	BAM_MODE_EN			BIT(0)
  
@@ -28,7 +28,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  /*
   * the NAND controller performs reads/writes with ECC in 516 byte chunks.
   * the driver calls the chunks 'step' or 'codeword' interchangeably
-@@ -430,6 +435,13 @@ struct qcom_nand_controller {
+@@ -429,6 +434,13 @@ struct qcom_nand_controller {
   * @cfg0, cfg1, cfg0_raw..:	NANDc register configurations needed for
   *				ecc/non-ecc mode for the current nand flash
   *				device
@@ -42,7 +42,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
   */
  struct qcom_nand_host {
  	struct nand_chip chip;
-@@ -452,6 +464,9 @@ struct qcom_nand_host {
+@@ -451,6 +463,9 @@ struct qcom_nand_host {
  	u32 ecc_bch_cfg;
  	u32 clrflashstatus;
  	u32 clrreadstatus;
@@ -52,7 +52,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  };
  
  /*
-@@ -460,12 +475,14 @@ struct qcom_nand_host {
+@@ -459,12 +474,14 @@ struct qcom_nand_host {
   * @ecc_modes - ecc mode for NAND
   * @is_bam - whether NAND controller is using BAM
   * @is_qpic - whether NAND CTRL is part of qpic IP
@@ -67,7 +67,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  	u32 dev_cmd_reg_start;
  };
  
-@@ -1604,7 +1621,7 @@ qcom_nandc_read_cw_raw(struct mtd_info *
+@@ -1603,7 +1620,7 @@ qcom_nandc_read_cw_raw(struct mtd_info *
  	data_size1 = mtd->writesize - host->cw_size * (ecc->steps - 1);
  	oob_size1 = host->bbm_size;
  
@@ -76,7 +76,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  		data_size2 = ecc->size - data_size1 -
  			     ((ecc->steps - 1) * 4);
  		oob_size2 = (ecc->steps * 4) + host->ecc_bytes_hw +
-@@ -1685,7 +1702,7 @@ check_for_erased_page(struct qcom_nand_h
+@@ -1684,7 +1701,7 @@ check_for_erased_page(struct qcom_nand_h
  	}
  
  	for_each_set_bit(cw, &uncorrectable_cws, ecc->steps) {
@@ -85,7 +85,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  			data_size = ecc->size - ((ecc->steps - 1) * 4);
  			oob_size = (ecc->steps * 4) + host->ecc_bytes_hw;
  		} else {
-@@ -1844,7 +1861,7 @@ static int read_page_ecc(struct qcom_nan
+@@ -1843,7 +1860,7 @@ static int read_page_ecc(struct qcom_nan
  	for (i = 0; i < ecc->steps; i++) {
  		int data_size, oob_size;
  
@@ -94,7 +94,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  			data_size = ecc->size - ((ecc->steps - 1) << 2);
  			oob_size = (ecc->steps << 2) + host->ecc_bytes_hw +
  				   host->spare_bytes;
-@@ -1941,6 +1958,30 @@ static int copy_last_cw(struct qcom_nand
+@@ -1940,6 +1957,30 @@ static int copy_last_cw(struct qcom_nand
  	return ret;
  }
  
@@ -125,7 +125,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  /* implements ecc->read_page() */
  static int qcom_nandc_read_page(struct nand_chip *chip, uint8_t *buf,
  				int oob_required, int page)
-@@ -1949,6 +1990,9 @@ static int qcom_nandc_read_page(struct n
+@@ -1948,6 +1989,9 @@ static int qcom_nandc_read_page(struct n
  	struct qcom_nand_controller *nandc = get_qcom_nand_controller(chip);
  	u8 *data_buf, *oob_buf = NULL;
  
@@ -135,7 +135,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  	nand_read_page_op(chip, page, 0, NULL, 0);
  	data_buf = buf;
  	oob_buf = oob_required ? chip->oob_poi : NULL;
-@@ -1968,6 +2012,9 @@ static int qcom_nandc_read_page_raw(stru
+@@ -1967,6 +2011,9 @@ static int qcom_nandc_read_page_raw(stru
  	int cw, ret;
  	u8 *data_buf = buf, *oob_buf = chip->oob_poi;
  
@@ -145,7 +145,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  	for (cw = 0; cw < ecc->steps; cw++) {
  		ret = qcom_nandc_read_cw_raw(mtd, chip, data_buf, oob_buf,
  					     page, cw);
-@@ -1988,6 +2035,9 @@ static int qcom_nandc_read_oob(struct na
+@@ -1987,6 +2034,9 @@ static int qcom_nandc_read_oob(struct na
  	struct qcom_nand_controller *nandc = get_qcom_nand_controller(chip);
  	struct nand_ecc_ctrl *ecc = &chip->ecc;
  
@@ -155,7 +155,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  	clear_read_regs(nandc);
  	clear_bam_transaction(nandc);
  
-@@ -2008,6 +2058,9 @@ static int qcom_nandc_write_page(struct
+@@ -2007,6 +2057,9 @@ static int qcom_nandc_write_page(struct
  	u8 *data_buf, *oob_buf;
  	int i, ret;
  
@@ -165,7 +165,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  	nand_prog_page_begin_op(chip, page, 0, NULL, 0);
  
  	clear_read_regs(nandc);
-@@ -2023,7 +2076,7 @@ static int qcom_nandc_write_page(struct
+@@ -2022,7 +2075,7 @@ static int qcom_nandc_write_page(struct
  	for (i = 0; i < ecc->steps; i++) {
  		int data_size, oob_size;
  
@@ -174,7 +174,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  			data_size = ecc->size - ((ecc->steps - 1) << 2);
  			oob_size = (ecc->steps << 2) + host->ecc_bytes_hw +
  				   host->spare_bytes;
-@@ -2080,6 +2133,9 @@ static int qcom_nandc_write_page_raw(str
+@@ -2079,6 +2132,9 @@ static int qcom_nandc_write_page_raw(str
  	u8 *data_buf, *oob_buf;
  	int i, ret;
  
@@ -184,7 +184,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  	nand_prog_page_begin_op(chip, page, 0, NULL, 0);
  	clear_read_regs(nandc);
  	clear_bam_transaction(nandc);
-@@ -2098,7 +2154,7 @@ static int qcom_nandc_write_page_raw(str
+@@ -2097,7 +2153,7 @@ static int qcom_nandc_write_page_raw(str
  		data_size1 = mtd->writesize - host->cw_size * (ecc->steps - 1);
  		oob_size1 = host->bbm_size;
  
@@ -193,7 +193,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  			data_size2 = ecc->size - data_size1 -
  				     ((ecc->steps - 1) << 2);
  			oob_size2 = (ecc->steps << 2) + host->ecc_bytes_hw +
-@@ -2158,6 +2214,9 @@ static int qcom_nandc_write_oob(struct n
+@@ -2157,6 +2213,9 @@ static int qcom_nandc_write_oob(struct n
  	int data_size, oob_size;
  	int ret;
  
@@ -203,7 +203,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  	host->use_ecc = true;
  	clear_bam_transaction(nandc);
  
-@@ -2806,6 +2865,7 @@ static int qcom_nand_host_init_and_regis
+@@ -2805,6 +2864,7 @@ static int qcom_nand_host_init_and_regis
  	struct nand_chip *chip = &host->chip;
  	struct mtd_info *mtd = nand_to_mtd(chip);
  	struct device *dev = nandc->dev;
@@ -211,7 +211,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  	int ret;
  
  	ret = of_property_read_u32(dn, "reg", &host->cs);
-@@ -2866,6 +2926,17 @@ static int qcom_nand_host_init_and_regis
+@@ -2865,6 +2925,17 @@ static int qcom_nand_host_init_and_regis
  	if (ret)
  		nand_cleanup(chip);
  
@@ -229,7 +229,7 @@ Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
  	return ret;
  }
  
-@@ -3032,6 +3103,7 @@ static int qcom_nandc_remove(struct plat
+@@ -3030,6 +3101,7 @@ static int qcom_nandc_remove(struct plat
  static const struct qcom_nandc_props ipq806x_nandc_props = {
  	.ecc_modes = (ECC_RS_4BIT | ECC_BCH_8BIT),
  	.is_bam = false,
diff --git a/target/linux/oxnas/patches-5.10/999-libata-hacks.patch b/target/linux/oxnas/patches-5.10/999-libata-hacks.patch
index 61c0c045a68e..84d418887c09 100644
--- a/target/linux/oxnas/patches-5.10/999-libata-hacks.patch
+++ b/target/linux/oxnas/patches-5.10/999-libata-hacks.patch
@@ -15,7 +15,7 @@
  	/* initialize internal qc */
  	qc = __ata_qc_from_tag(ap, ATA_TAG_INTERNAL);
  
-@@ -4554,6 +4562,9 @@ struct ata_queued_cmd *ata_qc_new_init(s
+@@ -4555,6 +4563,9 @@ struct ata_queued_cmd *ata_qc_new_init(s
  	if (unlikely(ap->pflags & ATA_PFLAG_FROZEN))
  		return NULL;
  
@@ -25,7 +25,7 @@
  	/* libsas case */
  	if (ap->flags & ATA_FLAG_SAS_HOST) {
  		tag = ata_sas_allocate_tag(ap);
-@@ -4599,6 +4610,8 @@ void ata_qc_free(struct ata_queued_cmd *
+@@ -4600,6 +4611,8 @@ void ata_qc_free(struct ata_queued_cmd *
  		qc->tag = ATA_TAG_POISON;
  		if (ap->flags & ATA_FLAG_SAS_HOST)
  			ata_sas_free_tag(tag, ap);
diff --git a/target/linux/ramips/patches-5.10/802-GPIO-MIPS-ralink-add-gpio-driver-for-ralink-SoC.patch b/target/linux/ramips/patches-5.10/802-GPIO-MIPS-ralink-add-gpio-driver-for-ralink-SoC.patch
index c173336924d2..52d40215f20d 100644
--- a/target/linux/ramips/patches-5.10/802-GPIO-MIPS-ralink-add-gpio-driver-for-ralink-SoC.patch
+++ b/target/linux/ramips/patches-5.10/802-GPIO-MIPS-ralink-add-gpio-driver-for-ralink-SoC.patch
@@ -62,7 +62,7 @@ Cc: linux-gpio@vger.kernel.org
  	depends on PLAT_SPEAR
 --- a/drivers/gpio/Makefile
 +++ b/drivers/gpio/Makefile
-@@ -119,6 +119,7 @@ obj-$(CONFIG_GPIO_PISOSR)		+= gpio-pisos
+@@ -120,6 +120,7 @@ obj-$(CONFIG_GPIO_PISOSR)		+= gpio-pisos
  obj-$(CONFIG_GPIO_PL061)		+= gpio-pl061.o
  obj-$(CONFIG_GPIO_PMIC_EIC_SPRD)	+= gpio-pmic-eic-sprd.o
  obj-$(CONFIG_GPIO_PXA)			+= gpio-pxa.o
diff --git a/target/linux/realtek/patches-5.10/002-5.13-gpio-add-realtek-otto-gpio-support.patch b/target/linux/realtek/patches-5.10/002-5.13-gpio-add-realtek-otto-gpio-support.patch
index fc1382428286..62e4338cbb27 100644
--- a/target/linux/realtek/patches-5.10/002-5.13-gpio-add-realtek-otto-gpio-support.patch
+++ b/target/linux/realtek/patches-5.10/002-5.13-gpio-add-realtek-otto-gpio-support.patch
@@ -56,7 +56,7 @@ Signed-off-by: Bartosz Golaszewski <bgolaszewski@baylibre.com>
  	help
 --- a/drivers/gpio/Makefile
 +++ b/drivers/gpio/Makefile
-@@ -124,6 +124,7 @@ obj-$(CONFIG_GPIO_RC5T583)		+= gpio-rc5t
+@@ -125,6 +125,7 @@ obj-$(CONFIG_GPIO_RC5T583)		+= gpio-rc5t
  obj-$(CONFIG_GPIO_RCAR)			+= gpio-rcar.o
  obj-$(CONFIG_GPIO_RDA)			+= gpio-rda.o
  obj-$(CONFIG_GPIO_RDC321X)		+= gpio-rdc321x.o
diff --git a/target/linux/realtek/patches-5.10/301-gpio-add-rtl8231-driver.patch b/target/linux/realtek/patches-5.10/301-gpio-add-rtl8231-driver.patch
index cdecb2a112e1..9175b55e8e10 100644
--- a/target/linux/realtek/patches-5.10/301-gpio-add-rtl8231-driver.patch
+++ b/target/linux/realtek/patches-5.10/301-gpio-add-rtl8231-driver.patch
@@ -15,7 +15,7 @@
  	depends on MFD_SYSCON
 --- a/drivers/gpio/Makefile
 +++ b/drivers/gpio/Makefile
-@@ -126,6 +126,7 @@ obj-$(CONFIG_GPIO_RDA)			+= gpio-rda.o
+@@ -127,6 +127,7 @@ obj-$(CONFIG_GPIO_RDA)			+= gpio-rda.o
  obj-$(CONFIG_GPIO_RDC321X)		+= gpio-rdc321x.o
  obj-$(CONFIG_GPIO_REALTEK_OTTO)		+= gpio-realtek-otto.o
  obj-$(CONFIG_GPIO_REG)			+= gpio-reg.o
